## Rating assembly

get reads together
cd /mnt/work
mkdir -p evaluation
cd evaluation

Because transrate only work on paired-end data, I attempted to use detonate, as can be seen below. I couldn't get it to function properly, and stopped trying. 

```
http://deweylab.biostat.wisc.edu/detonate/detonate-1.11.tar.gz
tar xvf detonate-1.11.tar.gz
cd detonate-1.11/
#make
NEED_CMAKE=yes make
```
```
# Run RSEM eval
#~/detonate-1.11/rsem-eval/rsem-eval-calculate-score -p 13 \
#                               --transcript-length-parameters ~/detonate-1.11/rsem-eval/rsem-eval-estimate-transcript-length-distribution \                               /mnt/work/assembly/cat_qual.fq \
#                               /mnt/work/assembly/trinity_out_dir/Trinity.fasta \
#                               Trinity_rsem_eval \
#                               101
```

Instead, I ran BUSCO to assess the completeness of the transcriptome. It did okay. 

*NB* I'm silly and named everything *nema* instead of thinking about it first. 

Run BUSCO for eukaryota
```
BUSCO.py \
  -i /mnt/work/assembly/trinity_out_dir/Trinity.fasta \
  -o nema_busco_eukaryota -l /home/ubuntu/busco/eukaryota_odb9 \
  -m tran --cpu 2
```
Check output
```
cat run_nema_busco_eukaryota/short_summary_nema_busco_eukaryota.txt
```
```
	C:4.6%[S:1.3%,D:3.3%],F:77.6%,M:17.8%,n:303

	14	Complete BUSCOs (C)
	4	Complete and single-copy BUSCOs (S)
	10	Complete and duplicated BUSCOs (D)
	235	Fragmented BUSCOs (F)
	54	Missing BUSCOs (M)
	303	Total BUSCO groups searched
```

Run BUSCO for plants
```
cd ~/busco
wget http://busco.ezlab.org/v2/datasets/embryophyta_odb9.tar.gz
tar -xzvf embryophyta_odb9.tar.gz
cd /mnt/work/evaluation
BUSCO.py \
  -i /mnt/work/assembly/trinity_out_dir/Trinity.fasta \
  -o nema_busco_embryophyta -l /home/ubuntu/busco/embryophyta_odb9 \
  -m tran --cpu 2
```

Run BUSCO for fungi
```
wget http://busco.ezlab.org/datasets/fungi_odb9.tar.gz
wget http://busco.ezlab.org/datasets/bacteria_odb9.tar.gz
/mnt/work/evaluation
BUSCO.py \
  -i /mnt/work/assembly/trinity_out_dir/Trinity.fasta \
  -o nema_busco_fungi -l /home/ubuntu/busco/fungi_odb9 \
  -m tran --cpu 2
```

Fungi output
```
	C:2.0%[S:0.3%,D:1.7%],F:67.2%,M:30.8%,n:290

	6	Complete BUSCOs (C)
	1	Complete and single-copy BUSCOs (S)
	5	Complete and duplicated BUSCOs (D)
	195	Fragmented BUSCOs (F)
	89	Missing BUSCOs (M)
	290	Total BUSCO groups searched
```

And, on the off chance some cute little bacteria got in...

Run BUSCO for bacteria
```
wget http://busco.ezlab.org/datasets/bacteria_odb9.tar.gz
BUSCO.py \
  -i /mnt/work/assembly/trinity_out_dir/Trinity.fasta \
  -o nema_busco_bacteria -l /home/ubuntu/busco/bacteria_odb9 \
  -m tran --cpu 2
 ``` 
Output
```
	C:2.0%[S:0.0%,D:2.0%],F:41.2%,M:56.8%,n:148

	3	Complete BUSCOs (C)
	0	Complete and single-copy BUSCOs (S)
	3	Complete and duplicated BUSCOs (D)
	61	Fragmented BUSCOs (F)
	84	Missing BUSCOs (M)
	148	Total BUSCO groups searched
```
