Assume hisat is already installed and olive reference genome has been indexed. 

Prepare for alignments
```
cd /mnt/work/
mkdir hisat
cd hisat
mkdir alignments
mkdir unaligned
```

Link in qc'd files
```
ln -s ../*/quality/*qc.fq.gz .
ls
```

Align single-end reads

Check folder locations before running
```
cd /mnt/work/hisat/
for f in SRR*
  do
    ~/hisat_bin/hisat2 -p 3 -x ~/hisat_index_olea_ref/olive_index -U ${f}  -S ./alignments/${f}.sam --met-file ./alignments/${f}.met --un-gz ./unaligned/unaligned_${f}.gz
done
```

Align paired-end reads
```
cd /mnt/work/hisat/

~/hisat_bin/hisat2 -p 2 -x ~/hisat_index_olea_ref/olive_index -1 ERR1346597_1.qc.fq.gz -2 ERR1346597_2.qc.fq.gz -S ./alignments/ERR1346597qc.fq.gz.sam --met-file ./alignments/ERR1346597.qc.fq.gz.met --un-gz ./unaligned/unaligned-se_ERR1346597qc.fq.gz --un-conc-gz ./unaligned/unaligned-pe_ERR1346597qc.fq.gz
```

Link in transcriptome
```
cd /mnt/work/hisat/alignments
ln -s ~/hisat_ref_genome/Oe6.scaffolds.fa
ls
```
Index transcriptome for samtools, convert to bam, sort bam, and index the sorted bam.
```
for i in *.sam
do
  samtools faidx Oe6.scaffolds.fa
  samtools view -bt Oe6.scaffolds.fa.fai ${i} > ${i}.bam
  samtools sort ${i}.bam -o ${i}.sorted.bam
  samtools index ${i}.sorted.bam 
done
```

Delete sam files, and unsorted bam files. 
