Assemble unaligned reads with megahit to see if microbial genomes come out of it. 

Install megahit
```
git clone https://github.com/voutcn/megahit.git
cd megahit
make -j 4
```

Run megahit on unaligned reads

Set up reads
```
cd /mnt/work/hisat/unaligned
mkdir megahit
cd megahit
ln -s ../*gz .
```
Run on paired end reads
```
~/megahit/megahit -1 unaligned-pe_ERR1346597_1.qc.fq.gz  -2 unaligned-pe_ERR1346597_2.qc.fq.gz  -m 0.9 -o megahit_ERR1346597
```

Remove the linked-in paired in reads
```
rm unaligned-pe_ERR1346597_1.qc.fq.gz unaligned-pe_ERR1346597_2.qc.fq.gz unaligned-se*
```

Run megahit on single end reads
```
for infile in *gz
do
  ~/megahit/megahit --r $infile -o ${infile}.megahit
done
```

The contigs were small (300bp to 5 or 6kbp), but in bacteria land, and especially in RNA seq land, that's not half bad. 

Possible next steps include something like prokka > salmon > kraken/kaiju with the contigs. 

Just running kraken/kaiju with contigs

Start with just running Kaiju because of its handy output and quick run time. Go back and do prokka and salmon if kaiju makes that seem like a worthwhile effort. 
