 Run salmon
 
 Oraganize files and set things up. Link in all reads and use `--libType A` for salmon to automatically infer what library type is present.
 ```
 cd /mnt/work/hisat/unaligned/
 mkdir salmon
 cd salmon
 ln -s ../*gz .
 ```
 Index the transcriptomes with salmon
 ```
 ln -s ../megahit/*megahit* .
 for infile in ./*/final.contigs.fa;
    do 
    j=$(dirname $infile)
    k=$(basename $j)
    salmon index --index ${k}_index --type quasi -k 31 --transcripts ${infile}
 done
 ```
 
 ```
 for infile in *.fq.gz
 do
     salmon quant -i ${infile}.megahit_index --libType A -r ${infile} -o ${infile}.quant --seqBias --gcBias
 done
