Install Circos

Install dependency
```
sudo apt-get -y install libgd-perl
```
Install relevant folder
```
cd ~
mkdir circos
cd circos
curl -O http://dib-training.ucdavis.edu.s3.amazonaws.com/metagenomics-scripps-2016-10-12/circos-0.69-3.tar.gz
tar -xvzf circos-0.69-3.tar.gz
```
Export circos path
```
export PATH=~/circos/circos-0.69-3/bin:$PATH
```
Define modules
```
circos -modules > modules
```
Install missing modules
```
grep missing modules |cut -f13 -d " " > missing_modules
for mod in $(cat missing_modules);
  do
  sudo cpan install $mod;
  done
```
Check module installation
```
circos -modules
```

Start plotting
```
cd /mnt/work/hisat/unaligned
mkdir circos
cd circos
```

Link in relevant files
```
ln -s /mnt/work/hisat/unaligned/prokka/*_prokka .
ln -s /mnt/work/hisat/unaligned/megahit/*megahit* .
ln -s /mnt/work/hisat/unaligned/salmon/*counts .
```

Grab some scripts and some contig files
```
curl -L -O https://github.com/ngs-docs/2016-metagenomics-sio/raw/master/circos-build.tar.gz
tar -xvzf circos-build.tar.gz
#curl -L -O https://s3-us-west-1.amazonaws.com/dib-training.ucdavis.edu/metagenomics-scripps-2016-10-12/subset_assembly.fa.gz
#gunzip subset_assembly.fa.gz
#mv subset_assembly.fa final.contigs.fa
```

Extract contigs above 1000 bp using khmer
```
for infile in ./*megahit*/final.contigs.fa
do
   j=$(dirname $infile)
   extract-long-sequences.py  $infile -l 1000 -o ${j}/final.contigs.long.fa
done
```

Cross my fingers and hope the parsing script works
```
python parse_data_for_circos.py
```

