There is a database from 454 pyrosequencing of the olive transcriptome. Download the sequences contained in that database and run `sbt_gather` to determine if any obvious fungal genes snuck in.


```
cd /mnt/work/ref_genome/
mkdir olea_estdb
cd olea_estdb
wget http://140.164.45.140/oleaestdb/download/Unique_transcripts/OLEEU.UT.fasta.gz
gunzip OLEEU.UT.fasta.gz
source ~/sourmashEnv2/bin/activate
```

Compute sourmash signature with --scaled 1000
```
sourmash compute -k 21,31,51 --scaled 1000 -o OLEUU.UT.fasta.sig OLEEU.UT.fasta
```

Run sbt_gather
```
sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json --threshold-bp=50000 -o fungi_k21_s1000_OLEUU.UT.fasta.sig.txt OLEUU.UT.fasta.sig
```
The output of sbt_gather was:
```
overlap    p_query p_genome
-------    ------- --------
62.0 kbp   0.4%      0.1%      Neosp1_AssemblyScaffolds.fasta.gz
found less than 28.0 kbp in common. => exiting

found 1 matches total;
the recovered matches hit 0.5% of the query
```

Make a blast database out of NeoSp1, and then blast Olea EST DB sequences against it
```
cp /mnt/work/blast/fungal_genomes/Neosp1_AssemblyScaffolds.fasta .
makeblastdb -in Neosp1_AssemblyScaffolds.fasta -parse_seqids -dbtype nucl

blastn -db Neosp1_AssemblyScaffolds.fasta -query OLEEU.UT.fasta -out OLEUU.UT-vs-NeoSp1.out -outfmt 11
```
Change format to human-readable
```
blast_formatter -archive OLEUU.UT-vs-NeoSp1.out -outfmt 6 -out OLEUU.UT-vs-NeoSp1.out.tab
blast_formatter -archive OLEUU.UT-vs-NeoSp1.out -outfmt 0 -out OLEUU.UT-vs-NeoSp1.out.pw
```

That wasn't terribly successful. Instead, try the find-hashes-2.py approach. Copy the NeoSp1 Signature into the folder, do find-hash-2.py, and then collect-reads.py. 
```
cp ~/fungal_sigs/Neosp1_AssemblyScaffolds.fasta.gz_s1k.sig .
python find-hashes.py -k 21 --min-samples 2 *sig
```
Output mat, mat.labels.txt. Ignore for now.

```
python find-hashes-2.py -k 21 *sig
```
Output savelist.txt

Feed into collect-reads.py
Tried, failed, try again later
```

