I got very whacky results from sbt_gather when only one signature was in the SBT. Use sourmash compare to estimate the similarity between the olive reference genome and olive fruit RNA-seq reads.
###### Note that these are preliminary results that will likely be off by 1-2% points, as there was a small bug in the code and the version of sourmash was not updated before running this. 

Link in the reference genome to the signatures folder
```
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa.sig .
```

Link in the signature reads to the compare folder
```
cd /mnt/work/sourmash
mkdir compare
cd compare
ln -s ../signatures/*sig .
```

Activate the environment and run sourmash
```
source sourmashEnv2/bin/activate
sourmash compare --ignore-abundance ./*sig
```

Returned problem with max hash, where max hash of Oe6 was 36893488147419104 and max hash of ERR\* was 1844674407370955. This is annoying. 

Recalculate signatures with abundances ignored. 

Link in fasta and fastq files
```
mkdir /mnt/work/sourmash/signatures2
cd signatures2
ln -s /mnt/work/sourmash/trim-low-abund/*abundtrim.gz .
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa.gz .
```

recalculate signatures
```
for infile in *gz
do
  sourmash compute -k 31 --scaled 10000 -o ${infile}.sig2 ${infile}
done
```

Activate the environment and run sourmash
```
source sourmashEnv2/bin/activate
sourmash compare --ignore-abundance ./*sig2
```

#### Results before updating sourmash
```
$ sourmash compare --ignore-abundance ./*sig2

# running sourmash subcommand: compare
loading ./ERR1346597.pe+orphans.qc.fq.gz.abundtrim.gz.sig2
loading ./Oe6.scaffolds.fa.gz.sig2
loading ./SRR1573503qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574328qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574719qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574772qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236909qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236910qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236911qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236912qc.fq.gz.abundtrim.gz.sig2
0-ERR1346597.pe+orphans.qc.fq.gz.abundtrim.gz   [ 1.     0.49   0.191  0.237  0.184  0.202  0.182  0.159  0.161  0.222]
1- Oe6.scaffolds.fa.gz                  [ 0.134  1.     0.046  0.058  0.043  0.049  0.044  0.038  0.039  0.055]
2-SRR1573503qc.fq.gz.abundtrim.gz       [ 0.363  0.316  1.     0.409  0.288  0.254  0.226  0.204  0.202  0.261]
3-SRR1574328qc.fq.gz.abundtrim.gz       [ 0.282  0.251  0.256  1.     0.205  0.184  0.165  0.146  0.146  0.193]
4-SRR1574719qc.fq.gz.abundtrim.gz       [ 0.368  0.312  0.304  0.346  1.     0.318  0.229  0.209  0.206  0.263]
5-SRR1574772qc.fq.gz.abundtrim.gz       [ 0.642  0.566  0.425  0.493  0.504  1.     0.385  0.347  0.339  0.443]
6-SRR4236909qc.fq.gz.abundtrim.gz       [ 0.59   0.525  0.386  0.451  0.37   0.392  1.     0.49   0.487  0.608]
7-SRR4236910qc.fq.gz.abundtrim.gz       [ 0.42   0.367  0.283  0.324  0.276  0.289  0.4    1.     0.464  0.471]
8-SRR4236911qc.fq.gz.abundtrim.gz       [ 0.381  0.335  0.252  0.291  0.244  0.252  0.356  0.415  1.     0.424]
9-SRR4236912qc.fq.gz.abundtrim.gz       [ 0.481  0.436  0.298  0.352  0.285  0.302  0.406  0.386  0.388  1.   ]
min similarity in matrix: 0.038103034272
```

#### Results after updating sourmash
```
$ sourmash compare --ignore-abundance ./*sig2

# running sourmash subcommand: compare
loading ./ERR1346597.pe+orphans.qc.fq.gz.abundtrim.gz.sig2
loading ./Oe6.scaffolds.fa.gz.sig2
loading ./SRR1573503qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574328qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574719qc.fq.gz.abundtrim.gz.sig2
loading ./SRR1574772qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236909qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236910qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236911qc.fq.gz.abundtrim.gz.sig2
loading ./SRR4236912qc.fq.gz.abundtrim.gz.sig2
0-ERR1346597.pe+orphans.qc.fq.gz.abundtrim.gz   [ 1.     0.118  0.143  0.148  0.14   0.182  0.162  0.13   0.128  0.179]
1- Oe6.scaffolds.fa.gz                  [ 0.118  1.     0.042  0.049  0.039  0.047  0.043  0.036  0.036  0.052]
2-SRR1573503qc.fq.gz.abundtrim.gz       [ 0.143  0.042  1.     0.187  0.174  0.189  0.166  0.134  0.126  0.162]
3-SRR1574328qc.fq.gz.abundtrim.gz       [ 0.148  0.049  0.187  1.     0.148  0.155  0.138  0.112  0.108  0.143]
4-SRR1574719qc.fq.gz.abundtrim.gz       [ 0.14   0.039  0.174  0.148  1.     0.242  0.165  0.135  0.126  0.159]
5-SRR1574772qc.fq.gz.abundtrim.gz       [ 0.182  0.047  0.189  0.155  0.242  1.     0.241  0.187  0.169  0.219]
6-SRR4236909qc.fq.gz.abundtrim.gz       [ 0.162  0.043  0.166  0.138  0.165  0.241  1.     0.283  0.259  0.322]
7-SRR4236910qc.fq.gz.abundtrim.gz       [ 0.13   0.036  0.134  0.112  0.135  0.187  0.283  1.     0.281  0.269]
8-SRR4236911qc.fq.gz.abundtrim.gz       [ 0.128  0.036  0.126  0.108  0.126  0.169  0.259  0.281  1.     0.254]
9-SRR4236912qc.fq.gz.abundtrim.gz       [ 0.179  0.052  0.162  0.143  0.159  0.219  0.322  0.269  0.254  1.   ]
min similarity in matrix: 0.0357523247716
```

Next steps:

+ Use a k size of 21 instead. With different cultivars, 31 may be too large.
+ Use `trim-low-abund` without `-v`, and rerun `sourmash compare` and `sbt_gather`
+ Use megahit to assemble unaligned hisat reads
