
Download SBT signatures for microbes

CSV for microbes in sequence bloom tree: https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/sra-bacteria-wgs-360k.categories.csv.gz
```
mkdir sourmash_SBTs
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k21-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k31-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k51-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k21-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k31-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k51-2016-11-27.tar.gz
```

Raw fungal genomes were collected by Luiz Irber. 

See: https://github.com/luizirber/2017-jgi-download

Download raw fungal genomes from Luiz
```
wget http://work.datasounds.org/microbial/fungi.tar.gz
tar -xvf fungi.tar.gz
```

Index fungal genomes
```
source sourmashEnv2/bin/activate
```
Get everything organized
```
cd ~/sourmash_SBTs
mkdir fungal_sigs
cd fungal_sigs
ln -s ../outputs/*gz .
```
Compute signatures for each genomes, beginning with a *kmer* size of 31 and sampling one in every 1000 kmers (*scaled*).
```
for infile in *.gz
  do
  sourmash compute -k 31 --scaled 1000 $infile
done
```
Index the signatures from the fungal genomes
```
sourmash sbt_index -k 31 KFG_3.27.17 *sig
```

Now, download all fungal assemblies from genbank
Create a virtual environment
```
cd ~
python2.7 -m virtualenv ncbi-genome-downloadEnv
source ncbi-genome-downloadEnv/bin/activate
cd ncbi-genome-downloadEnv
git clone https://github.com/kblin/ncbi-genome-download.git
cd ncbi-genome-download
pip install .
source ncbi-genome-downloadEnv/bin/activate
```

```
ncbi-genome-download -s genbank -l all -F fasta fungi
```
Compute signature for index
```
for infile in ./*/GCA*
  do
  sourmash compute -k 31 --scaled 1000 $infile
done
```

