
Download SBT signatures for microbes

CSV for microbes in sequence bloom tree: https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/sra-bacteria-wgs-360k.categories.csv.gz
```
mkdir sourmash_SBTs
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k21-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k31-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k51-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k21-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k31-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k51-2016-11-27.tar.gz
```

Raw fungal genomes were collected by Luiz Irber. 

See: https://github.com/luizirber/2017-jgi-download

Download raw fungal genomes from Luiz
```
mkdir ~/fungal_genomes
cd ~/fungal_genomes
mkdir JGI_1KFGs
cd JGI_1KFGs
wget http://work.datasounds.org/microbial/fungi.tar.gz
tar -xvf fungi.tar.gz
```

Source sourmash environment
```
source sourmashEnv2/bin/activate
```
Get everything organized--make a directory that all fungal signature files will go in to, and link in the JGI 1KFG samples. 
```
cd ~
mkdir fungal_sigs
cd fungal_sigs
ln -s ~fungal_genomes/JGI_1KFGs/*gz .
```
Compute signatures for each JGI 1K FG genome, beginning with a *kmer* size of 31 and sampling one in every 1000 kmers (*scaled*).
```
for infile in *.gz
  do
  sourmash compute -k 31 --scaled 1000 $infile
done
```
Now, download all fungal assemblies from genbank and refseq. To do so, download `ncbi-genome-download` tool. 

Create a virtual environment
```
cd ~
python2.7 -m virtualenv ncbi-genome-downloadEnv
source ncbi-genome-downloadEnv/bin/activate
cd ncbi-genome-downloadEnv
git clone https://github.com/kblin/ncbi-genome-download.git
cd ncbi-genome-download
pip install .
source ncbi-genome-downloadEnv/bin/activate
```
Download the genomes from genbank and refseq
```
cd ~/fungal_genomes
mkdir ncbi-genome-downloads-fungi
cd ncbi-genome-downloads-fungi
ncbi-genome-download -s genbank -l all -F fasta fungi
ncbi-genome-download -s refseq -l all -F fasta fungi
```
Compute signature for index
```
for infile in ./*/*/GC*/*.gz
  do
  sourmash compute -k 31 --scaled 1000 $infile
done
```
Move signature files to `fungal_sigs` folder
```
for infile in *sig
do
  mv $infile ~/fungal_sigs/$infile
done
```

Index the signatures from all fungal genomes. First, link `.sig` files in to the `sourmash_SBTs` folder, then index, then remove the linked files. 

Note 3328 files were loaded in to SBT. 
```
cd ~/sourmash_SBTs
ln -s ../fungal_sigs/*sig .
sourmash sbt_index -k 31 fungal_4.08.17 *sig
rm *sig
```

