
Download SBT signatures for microbes

CSV for microbes in sequence bloom tree: https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/sra-bacteria-wgs-360k.categories.csv.gz
```
mkdir sourmash_SBTs
cd sourmash_SBTs
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k21-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k31-2016-11-27.tar.gz
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k51-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k21-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k31-2016-11-27.tar.gz
tar -xzvf microbe-sbt-k51-2016-11-27.tar.gz
```

Raw fungal genomes were collected by Luiz Irber. 

See: https://github.com/luizirber/2017-jgi-download

Download raw fungal genomes from Luiz
```
mkdir ~/fungal_genomes
cd ~/fungal_genomes
mkdir JGI_1KFGs
cd JGI_1KFGs
wget http://work.datasounds.org/microbial/fungi.tar.gz
tar -xvf fungi.tar.gz
```

Source sourmash environment
```
source sourmashEnv2/bin/activate
```
Get everything organized--make a directory that all fungal signature files will go in to, and link in the JGI 1KFG samples. 
```
cd ~
mkdir fungal_sigs
```
Compute signatures for each JGI 1K FG genome, beginning with a *kmer* size of 31 and sampling one in every 10000 kmers (*scaled*).
```
cd fungal_sigs
for infile in *gz
  do
  sourmash compute -k 31 --scaled 10000 $infile
done
```

Now, download all fungal assemblies from genbank and refseq. To do so, download `ncbi-genome-download` tool. 

Create a virtual environment
```
cd ~
python2.7 -m virtualenv ncbi-genome-downloadEnv
source ncbi-genome-downloadEnv/bin/activate
cd ncbi-genome-downloadEnv
git clone https://github.com/kblin/ncbi-genome-download.git
cd ncbi-genome-download
pip install .
source ncbi-genome-downloadEnv/bin/activate
```
Download the genomes from genbank and refseq
```
cd ~/fungal_genomes
mkdir ncbi-genome-downloads-fungi
cd ncbi-genome-downloads-fungi
ncbi-genome-download -s genbank -l all -F fasta fungi
ncbi-genome-download -s refseq -l all -F fasta fungi
```
Compute signature for index
```
cd fungal_sigs
for infile in ~/fungal_genomes/ncbi-genome-downloads-fungi/*/*/GC*/*.gz
  do
  sourmash compute -k 31 --scaled 10000 --name-from-first $infile
done
```

Index the signatures from all fungal genomes. First, link `.sig` files in to the `sourmash_SBTs` folder, then index, then remove the linked files. 

4/13/17: Note 3328 files were loaded in to SBT. 
```
cd ~/sourmash_SBTs
ln -s ../fungal_sigs/*sig .
sourmash sbt_index -k 31 fungal_4.13.17 *sig
rm *sig
```

4/26/17: Recalculate with a `--scaled` of 1000, which will match the scaled value of the olive reference genome and the RNA-seq signatures labeled `*sig3`.

4/26/17: Compute signatures again, this time with `--scaled 1000`, and with all three k values
```
cd fungal_sigs
mkdir scaled_1k_sigs
ln -s  ~/fungal_genomes/JGI_1KFGs/*gz .
for infile in *gz
  do
  sourmash compute -k 21,31,51 --scaled 1000 -o ${infile}_s1k.sig $infile 
done
```

4/26/17: Compute signatures again, this time with `--scaled 1000` and with k 21, 31, 51
```
cd fungal_sigs/scaled_1k_sigs
for infile in ~/fungal_genomes/ncbi-genome-downloads-fungi/*/*/GC*/*.gz
  do
  sourmash compute -k 21,31,51 --scaled 1000 --name-from-first $infile
done
```
4/27/17: Index the new signatures. Make an SBT for k21, k31, k51. Label the SBT folder and the SBT with the scaled value used. As above, 3328 signatures were included in each SBT. 
k21:
```
cd ~/sourmash_SBTs/
mkdir fungal_k21_s1000 
cd fungal_k21_s1000 
ln -s ../../fungal_sigs/scaled_1k_sigs/*sig .
sourmash sbt_index -k 21 fungal_k21_s1000 *sig
rm *sig
```
k31:
```
cd ~/sourmash_SBTs/
mkdir fungal_k31_s1000 
cd fungal_k31_s1000 
ln -s ../../fungal_sigs/scaled_1k_sigs/*sig .
sourmash sbt_index -k 31 fungal_k31_s1000 *sig
rm *sig
```
k51:
```
cd ~/sourmash_SBTs/
mkdir fungal_k51_s1000 
cd fungal_k51_s1000 
ln -s ../../fungal_sigs/scaled_1k_sigs/*sig .
sourmash sbt_index -k 51 fungal_k51_s1000 *sig
rm *sig
```


And last but not least, download and index the reference genome to provide an estimate of the amoung of olive genome in the sample vs. the amount of microbe genome in the sample.

Although sourmash sbt_gather is awesome at identifying things that are in the SBT, it can't identify things that are not. It is important to recognize the microbial fraction in the olive RNA-seq reads in order to estimate the potential for microbial interactions in post-harvest processes.

Use reference genome downloaded here: http://denovo.cnag.cat/genomes/olive/download/Oe6/Oe6.scaffolds.fa.gz

Use the olive signature used to estimate contamination in the sample (https://github.com/taylorreiter/olive_public_seq/blob/master/RNA-seq/many_fruit/sourmash/2.ref_genome_contamination_sourmash.md)


```
cd ~/sourmash_SBTs
mkdir olive_ref
cd olive_ref
sourmash sbt_index -k 31 olive_ref_4.18.17 /mnt/work/ref_genome/*sig
```

To time things:
```
/usr/bin/time -v -o time_out.txt -- sourmash...
```

5/15/17

Downsample the `--scaled 1000` signatures to 10000 for -k 21,31,51. Remake the index. Re-run sourmash and compare the results. Run on the reference genome and on the RNA-seq reads. 

Organize files
```
cd ~/fungal_sigs/
mkdir downsample_s10000
cd downsample_s10000
ln -s ../scaled_1k_sigs/*s1k.sig .
```

Download the `pick-k-and-downsample.py` script from https://github.com/ctb/tara-sourmash/blob/master/pick-k-and-downsample.py
File not doing the thing that I want, change it to be `-k 21`. 

```
for infile in *s1k.sig
do
  j=$(basename $infile _s1k.sig)
  python pick-k-and-downsample.py $infile > ${j}_s10k.sig
done
```

Check the length of files:
```
wc -l Zymps1_AssemblyScaffolds.fasta.gz_s1k.sig 
# 94384 Zymps1_AssemblyScaffolds.fasta.gz_s1k.sig
wc -l Zymps1_AssemblyScaffolds.fasta.gz_s10k.sig 
# 3209 Zymps1_AssemblyScaffolds.fasta.gz_s10k.sig
```

Do again with non-jgi sigs
```
ln -s ../scaled_1k_sigs/*gz.sig .
for infile in *gz.sig
do
  j=$(basename $infile gz.sig)
  python pick-k-and-downsample.py $infile > ${j}_s10k.sig
done
```

Make a new index
```
cd ~/fungal_sigs/downsample_s10000
rm *s1k.sig

rm *gz.sig

cd ~/sourmash_SBTs/
mkdir fungal_k21_s10k
cd fungal_k21_s10k
ln -s ../../fungal_sigs/downsample_s10000/*sig .
sourmash sbt_index -k 21 fungal_k21_s10k *sig
rm *sig
```

-------------------------------------
5/16/17

Remove old microbe sbts that were downloaded above.
Download new microbe sbts, and make sure to place them in a folder to make them easier to move around. 
```
cd ~/sourmash_SBTs
mkdir refseq_k21_17-5-9
cd refseq_k21_17-5-9/
wget https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/microbe-refseq-sbt-k21-2017.05.09.tar.gz


cd ~/sourmash_SBTs
mkdir genbank_k21_17-5-9
cd genbank_k21_17-5-9
wget https://s3-us-west-1.amazonaws.com/spacegraphcats.ucdavis.edu/microbe-genbank-sbt-k21-2017.05.09.tar.gz
rm microbe-genbank-sbt-k21-2017.05.09.tar.gz 
```
-----------------------------------
All of refseq and genbank is pretty slow. Re-download 50k microbes k21, run against olive reference genome
```
cd /mnt/work
mkdir other_sbts
cd other_sbts
mkdir microbes_k21
cd microbes_k21
wget http://spacegraphcats.ucdavis.edu.s3.amazonaws.com/microbe-sbt-k21-2016-11-27.tar.gz
tar xf microbe-sbt-k21-2016-11-27.tar.gz
```

