Some fungal hashes were identified from unaligned hisat reads. See if there is contamination in the reference genome from fungal genomes that are in the 1000KFG index. 

Begin by downloading the reference genome and computing a signature on it. 
```
mkdir /mnt/work/ref_genome
cd ref_genome
wget http://denovo.cnag.cat/genomes/olive/download/Oe6/Oe6.scaffolds.fa.gz
source ~/sourmashEnv2/bin/activate
sourmash compute -k 21,31,51 -o Oe6.scaffolds.fa.sig --scaled 500 Oe6.scaffolds.fa.gz
```
using fungal index:
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_sigs/KFG_3.27.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.001 -o KFG_k31_Oe6.scaffolds.fa.txt
```

Output, where the first column is fraction of genome found, and the second is fraction of query metagenome that is the genome
```
# running sourmash subcommand: sbt_gather
loaded query: Oe6.scaffolds.fa.gz... (k=31, DNA)
query signature has max_hash: 36893488147419104
found: 0.00 0.00 Neosp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Melli1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Anasp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Ophpic1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
found: 0.00 0.00 Croqu1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 PirE2_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Cunech1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Calful1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Pirfi3_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 EndpusZ1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Pilano1_AssemblyScaffolds.fasta.gz
found 14, total fraction 0.000

Composition:
0.00 Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
0.00 Neosp1_AssemblyScaffolds.fasta.gz
0.00 Ophpic1_AssemblyScaffolds.fasta.gz
0.00 Anasp1_AssemblyScaffolds.fasta.gz
0.00 Cunech1_AssemblyScaffolds.fasta.gz
0.00 Melli1_AssemblyScaffolds.fasta.gz
0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
0.00 Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
0.00 PirE2_1_AssemblyScaffolds.fasta.gz
0.00 Calful1_AssemblyScaffolds.fasta.gz
0.00 Croqu1_AssemblyScaffolds.fasta.gz
0.00 EndpusZ1_AssemblyScaffolds.fasta.gz
0.00 Pilano1_AssemblyScaffolds.fasta.gz
0.00 Pirfi3_AssemblyScaffolds.fasta.gz
```

Do this again, this time using the larger index that incorporates genbank and refseq as well as JGI 1KFG and a higher threshold. Note this index has a scaled of 1000. 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.08.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```

Intentionally stopped early bc of odd file names, but preliminary output:
```
# running sourmash subcommand: sbt_gather
loaded query: Oe6.scaffolds.fa.gz... (k=31, DNA)
query signature has max_hash: 36893488147419104
found: 0.00 0.00 Neosp1_AssemblyScaffolds.fasta.gz
found: 0.01 0.00 ./genbank/fungi/GCA_001678115.1/GCA_001678115.1_ASM167811v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001443065.1/GCA_001443065.1_ASM144306v1_genomic.fna.gz
found: 0.00 0.00 Anasp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001914275.1/GCA_001914275.1_FSWF8-bin-4_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001676635.1/GCA_001676635.1_ASM167663v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001593145.1/GCA_001593145.1_ASM159314v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_000286035.1/GCA_000286035.1_ASM28603v1_genomic.fna.gz
```
The index used was constructed at a scale of 500, and gave these results:
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```
```
Composition:
0.01 0.00 LVWM01000001.1 Aureobasidium pullulans isolate Santander contig_1, whole genome shotgun sequence
0.00 0.00 KQ503367.1 Puccinia psidii isolate Aus_3 unplaced genomic scaffold scaffold_1, whole genome shotgun sequence
0.00 0.00 /home/ubuntu/fungal_genomes/JGI_1KFGs/Neosp1_AssemblyScaffolds.fasta.gz
0.00 0.00 MSDY01000045.1 Aureobasidium sp. FSWF8-4 FSW8-bin-4-scaffold_100, whole genome shotgun sequence
0.00 0.00 /home/ubuntu/fungal_genomes/JGI_1KFGs/Croqu1_AssemblyScaffolds.fasta.gz
```
However there was a bug that prevented this result from being saved to the above file. 

This was ran again with an updated branch of sourmash (2017-metagenome). 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```
And the output was:

(note I shortened the JGI names here because they were annoyingly long)
```
cat fungi_k31_Oe6.scaffolds.fa.txt 
intersect_bp,f_orig_query,f_found_genome,name
260000.0 0.0019252128841169937 0.0002837808338790657 Neosp1_AssemblyScaffolds.fasta.gz
220000.0 0.0023905248288601543 0.00024012224405151714 KQ503367.1 Puccinia psidii isolate Aus_3 unplaced genomic scaffold scaffold_1
160000.0 0.008849557522123894 0.0001746343593101943 LVWM01000001.1 Aureobasidium pullulans isolate Santander 
80000.0 0.0009665571235260004,8.731717965509715e-05 Croqu1_AssemblyScaffolds.fasta.gz
140000.0 0.001751927119831815 0.00015280506439642 MSDY01000045.1 Aureobasidium sp. FSWF8-4 FSW8-bin-4-scaffold_100
70000.0 0.002135611318739989 7.640253219821e-05 CAGI01000713.1 Ustilago hordei Uh4857-4 WGS project CAGI00000000 data, contig UHOR_scaffold_4.01268
160000.0 0.00047969299648225136 0.0001746343593101943 Anasp1_AssemblyScaffolds.fasta.gz
150000.0 0.0005361930294906167 0.00016371971185330714 HF939678.1 Blumeria graminis f. sp. hordei DH14 genomic scaffold, sca001561
70000.0 0.00102880658436214 7.640253219821e-05 Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
140000.0 0.0007434944237918215 0.00015280506439642 PirE2_1_AssemblyScaffolds.fasta.gz
```

Then, use microbes index of k31 (output from 2017-metagenome branch). Time the process as it seems to be taken an inordinately long time.
```
/usr/bin/time -v -o time_microbes_k31.txt -- sourmash sbt_gather -k 31 ~/sourmash_SBTs/microbes.k31.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k31_Oe6.scaffolds.fa.txt
```

Stopped prematurely to update sourmash, but this was found
```
loaded query: Oe6.scaffolds.fa.gz... (k=21, DNA)
query signature has max_hash: 36893488147419104
found: 0.09 0.00 NC_008513.1 Buchnera aphidicola BCc, complete genome
```
And as a less sensitive detection limit use k21 for microbes as well, with new branch of sourmash and timed
```
/usr/bin/time -v -o time_microbes_k21.txt sourmash sbt_gather -k 21 ~/sourmash_SBTs/microbes.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k21_Oe6.scaffolds.fa.txt
```

4/26/17

I then moved all of these results into a folder indicating that all were for the olive reference genome signature with `--scaled 500`.
```
cd /mnt/work/ref_genome
mkdir scaled_500

for infile in *txt
do
  mv $infile ./scaled_500/$infile
done

mv Oe6.scaffolds.fa.sig scaled_500/Oe6.scaffolds.fa.scaled500.sig
```

I was skeptical of the behaviour, and the time that it was taking to make things work, so I re-indexed the fungal genomes with a `--scaled 1000` and `k 21,31,51`, and then re-ran things. I then re-calculated the olive reference genome signature with a `--scaled 1000` and `-k 21,31,51`, and then re-ran sbt_gather. I also set the bp detection of sbt_gather to 0kb. 

Begin by calculating a new signature for the olive reference genome
```
cd /mnt/work/ref_genome/
mkdir scaled_1000
cd scaled_1000
source ~/sourmashEnv2/bin/activate

sourmash compute -k 21,31,51 --scaled 1000 -o Oe6.scaffolds.fa.scaled1000.sig ../Oe6.scaffolds.fa.gz
```

Run sbt_gather
```
cd /mnt/work/ref_genome/scaled_1000
/usr/bin/time -v -o time_fungi_k21_s1000.txt -- sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k21__s1000_Oe6.scaffolds.fa.txt

/usr/bin/time -v -o time_fungi_k31_s1000.txt -- sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_k31_s1000/fungal_k31_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k31__s1000_Oe6.scaffolds.fa.txt

/usr/bin/time -v -o time_fungi_k51_s1000.txt -- sourmash sbt_gather -k 51 ~/sourmash_SBTs/fungal_k51_s1000/fungal_k51_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k51__s1000_Oe6.scaffolds.fa.txt
```
