Some fungal hashes were identified from unaligned hisat reads. See if there is contamination in the reference genome from fungal genomes that are in the 1000KFG index. 

Begin by downloading the reference genome and computing a signature on it. 
```
mkdir /mnt/work/ref_genome
cd ref_genome
wget http://denovo.cnag.cat/genomes/olive/download/Oe6/Oe6.scaffolds.fa.gz
source ~/sourmashEnv2/bin/activate
sourmash compute -k 21,31,51 -o Oe6.scaffolds.fa.sig --scaled 500 Oe6.scaffolds.fa.gz
```
using fungal index:
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_sigs/KFG_3.27.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.001 -o KFG_k31_Oe6.scaffolds.fa.txt
```

Output, where the first column is fraction of genome found, and the second is fraction of query metagenome that is the genome
```
# running sourmash subcommand: sbt_gather
loaded query: Oe6.scaffolds.fa.gz... (k=31, DNA)
query signature has max_hash: 36893488147419104
found: 0.00 0.00 Neosp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Melli1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Anasp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Ophpic1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
found: 0.00 0.00 Croqu1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 PirE2_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Cunech1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Calful1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Pirfi3_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 EndpusZ1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Pilano1_AssemblyScaffolds.fasta.gz
found 14, total fraction 0.000

Composition:
0.00 Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
0.00 Neosp1_AssemblyScaffolds.fasta.gz
0.00 Ophpic1_AssemblyScaffolds.fasta.gz
0.00 Anasp1_AssemblyScaffolds.fasta.gz
0.00 Cunech1_AssemblyScaffolds.fasta.gz
0.00 Melli1_AssemblyScaffolds.fasta.gz
0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
0.00 Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
0.00 PirE2_1_AssemblyScaffolds.fasta.gz
0.00 Calful1_AssemblyScaffolds.fasta.gz
0.00 Croqu1_AssemblyScaffolds.fasta.gz
0.00 EndpusZ1_AssemblyScaffolds.fasta.gz
0.00 Pilano1_AssemblyScaffolds.fasta.gz
0.00 Pirfi3_AssemblyScaffolds.fasta.gz
```

Do this again, this time using the larger index that incorporates genbank and refseq as well as JGI 1KFG and a higher threshold. Note this index has a scaled of 1000. 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.08.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```

Intentionally stopped early bc of odd file names, but preliminary output:
```
# running sourmash subcommand: sbt_gather
loaded query: Oe6.scaffolds.fa.gz... (k=31, DNA)
query signature has max_hash: 36893488147419104
found: 0.00 0.00 Neosp1_AssemblyScaffolds.fasta.gz
found: 0.01 0.00 ./genbank/fungi/GCA_001678115.1/GCA_001678115.1_ASM167811v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001443065.1/GCA_001443065.1_ASM144306v1_genomic.fna.gz
found: 0.00 0.00 Anasp1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 Orpsp1_1_AssemblyScaffolds.fasta.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001914275.1/GCA_001914275.1_FSWF8-bin-4_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001676635.1/GCA_001676635.1_ASM167663v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_001593145.1/GCA_001593145.1_ASM159314v1_genomic.fna.gz
found: 0.00 0.00 ./genbank/fungi/GCA_000286035.1/GCA_000286035.1_ASM28603v1_genomic.fna.gz
```
The index used was constructed at a scale of 500, and gave these results:
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```
```
Composition:
0.01 0.00 LVWM01000001.1 Aureobasidium pullulans isolate Santander contig_1, whole genome shotgun sequence
0.00 0.00 KQ503367.1 Puccinia psidii isolate Aus_3 unplaced genomic scaffold scaffold_1, whole genome shotgun sequence
0.00 0.00 /home/ubuntu/fungal_genomes/JGI_1KFGs/Neosp1_AssemblyScaffolds.fasta.gz
0.00 0.00 MSDY01000045.1 Aureobasidium sp. FSWF8-4 FSW8-bin-4-scaffold_100, whole genome shotgun sequence
0.00 0.00 /home/ubuntu/fungal_genomes/JGI_1KFGs/Croqu1_AssemblyScaffolds.fasta.gz
```

Then, use microbes index of k31
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/microbes.k31.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k31_Oe6.scaffolds.fa.txt
```



And as a less sensitive detection limit use k21 for microbes as well
```
sourmash sbt_gather -k 21 ~/sourmash_SBTs/microbes.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k21_Oe6.scaffolds.fa.txt
```
