Some fungal hashes were identified from unaligned hisat reads. See if there is contamination in the reference genome from fungal genomes that are in the 1000KFG index. 

Begin by downloading the reference genome and computing a signature on it. 
```
mkdir /mnt/work/ref_genome
cd ref_genome
wget http://denovo.cnag.cat/genomes/olive/download/Oe6/Oe6.scaffolds.fa.gz
source ~/sourmashEnv2/bin/activate
sourmash compute -k 21,31,51 -o Oe6.scaffolds.fa.sig --scaled 500 Oe6.scaffolds.fa.gz
```
using fungal index:
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_sigs/KFG_3.27.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.001 -o KFG_k31_Oe6.scaffolds.fa.txt
```

Do this again, this time using the larger index that incorporates genbank and refseq as well as JGI 1KFG and a higher threshold. Note this index has a scaled of 1000. 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.08.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```

The index used was constructed at a scale of 500. Bug that prevented this result from being saved to the above file. 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```

This was ran again with an updated branch of sourmash (2017-metagenome). 
```
sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o fungi_k31_Oe6.scaffolds.fa.txt
```

Then, use microbes index of k31 (output from 2017-metagenome branch). Time the process as it seems to be taken an inordinately long time.
```
/usr/bin/time -v -o time_microbes_k31.txt -- sourmash sbt_gather -k 31 ~/sourmash_SBTs/microbes.k31.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k31_Oe6.scaffolds.fa.txt
```

Stopped prematurely to update sourmash, but this was found
```
loaded query: Oe6.scaffolds.fa.gz... (k=21, DNA)
query signature has max_hash: 36893488147419104
found: 0.09 0.00 NC_008513.1 Buchnera aphidicola BCc, complete genome
```
And as a less sensitive detection limit use k21 for microbes as well, with new branch of sourmash and timed
```
/usr/bin/time -v -o time_microbes_k21.txt sourmash sbt_gather -k 21 ~/sourmash_SBTs/microbes.sbt.json Oe6.scaffolds.fa.sig --threshold=0.01 -o microbes_k21_Oe6.scaffolds.fa.txt
```

4/26/17

I then moved all of these results into a folder indicating that all were for the olive reference genome signature with `--scaled 500`.
```
cd /mnt/work/ref_genome
mkdir scaled_500

for infile in *txt
do
  mv $infile ./scaled_500/$infile
done

mv Oe6.scaffolds.fa.sig scaled_500/Oe6.scaffolds.fa.scaled500.sig
```

I was skeptical of the behaviour, and the time that it was taking to make things work, so I re-indexed the fungal genomes with a `--scaled 1000` and `k 21,31,51`, and then re-ran things. I then re-calculated the olive reference genome signature with a `--scaled 1000` and `-k 21,31,51`, and then re-ran sbt_gather. I also set the bp detection of sbt_gather to 0kb. 

Begin by calculating a new signature for the olive reference genome
```
cd /mnt/work/ref_genome/
mkdir scaled_1000
cd scaled_1000
source ~/sourmashEnv2/bin/activate

sourmash compute -k 21,31,51 --scaled 1000 -o Oe6.scaffolds.fa.scaled1000.sig ../Oe6.scaffolds.fa.gz
```

Run `sbt_gather`
```
cd /mnt/work/ref_genome/scaled_1000
/usr/bin/time -v -o time_fungi_k21_s1000.txt -- sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k21__s1000_Oe6.scaffolds.fa.txt

/usr/bin/time -v -o time_fungi_k31_s1000.txt -- sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_k31_s1000/fungal_k31_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k31_s1000_Oe6.scaffolds.fa.txt

/usr/bin/time -v -o time_fungi_k51_s1000.txt -- sourmash sbt_gather -k 51 ~/sourmash_SBTs/fungal_k51_s1000/fungal_k51_s1000.sbt.json Oe6.scaffolds.fa.scaled1000.sig --threshold-bp=0 -o fungi_k51_s1000_Oe6.scaffolds.fa.txt
```

**To Do**
Run sbt_gather again with k 21, but with the microbes index. While the fungal index was created at a --scaled of 1000, the microbes one is at a --scaled of 10000. 
