sourmash sbt_gather identified contamination in the olive reference genome. Set up BLAST database and see what happens!

Begin by making a blast database of the reference genome. Download the predicted proteins from the identified genomes and blast against reference genome. Get the sequences of interest from the reference genome and blast or align the nucelotides against the identified genome. Or something.

Making blast database:

Install blast
```
cd ~
sudo apt-get install ncbi-blast+
```

Unzip the reference genome to make blast happy
```
cd /mnt/work/ref_genome
gunzip Oe6.scaffolds.fa.gz
```
Make database of reference genome (note the db will be in the ref_genome folder)
```
cd /mnt/work/
mkdir blast
cd blast
makeblastdb -in ../ref_genome/Oe6.scaffolds.fa -parse_seqids -dbtype nucl
```

download fungal genome amino acids
```
mkdir fungal_genomes_aa
cd fungal_genomes_aa
git clone https://github.com/glarue/jgi-query.git
cd jgi-query
python jgi_query.py -c 
# enter mycocosm credentials. note that it must be entered between ''. 
```

Above attempt failed. Try to do a blast with one genome against another
```
cd mnt/work/blast/
mv fungal_genomes_aa/ fungal_genomes/
cd fungal_genomes
cp ~/fungal_genomes/JGI_1KFGs/Neosp1_AssemblyScaffolds.fasta.gz .
blastn -db ../../work/ref_genome/Oe6.scaffolds.fa -query Neosp1_AssemblyScaffolds.fasta -out Neosp1_n.out
```

