# BLAST
sourmash sbt_gather identified contamination in the olive reference genome. Set up BLAST database and see what happens!

Begin by making a blast database of the reference genome. Download the predicted proteins from the identified genomes and blast against reference genome. Get the sequences of interest from the reference genome and blast or align the nucelotides against the identified genome. Or something.

## Making blast database:

Install blast
```
cd ~
sudo apt-get install ncbi-blast+
```

Unzip the reference genome to make blast happy
```
cd /mnt/work/ref_genome
gunzip Oe6.scaffolds.fa.gz
```
Make database of reference genome (note the db will be in the ref_genome folder)
```
cd /mnt/work/
mkdir blast
cd blast
makeblastdb -in ../ref_genome/Oe6.scaffolds.fa -parse_seqids -dbtype nucl
```

download fungal genome amino acids
```
mkdir fungal_genomes_aa
cd fungal_genomes_aa
git clone https://github.com/glarue/jgi-query.git
cd jgi-query
python jgi_query.py -c 
# enter mycocosm credentials. note that it must be entered between ''. 
```

Above attempt failed. Try to do a blast with one genome against another
```
cd mnt/work/blast/
mv fungal_genomes_aa/ fungal_genomes/
cd fungal_genomes
cp ~/fungal_genomes/JGI_1KFGs/Neosp1_AssemblyScaffolds.fasta.gz .
blastn -db ../../work/ref_genome/Oe6.scaffolds.fa -query Neosp1_AssemblyScaffolds.fasta -out Neosp1_n.out -outfmt 6
```
Worked! not too slow. Upon seeing results, making a blastdb of fungi of interest, and going to blast olive reference genome to it.
```
mkdir sbt_fungi_blastdb
cd sbt_fungi_blastdb
```
Make a document with the JGI sequences in i:
```
nano jgi_fungi_ref_genome_k21.txt
```
Add:
```
Neosp1_AssemblyScaffolds.fasta.gz
Orpsp1_1_AssemblyScaffolds.fasta.gz
Anasp1_AssemblyScaffolds.fasta.gz
Gloin1_AssemblyScaffolds.fasta.gz
Pirfi3_AssemblyScaffolds.fasta.gz
Zoprh1_AssemblyScaffolds.fasta.gz
PirE2_1_AssemblyScaffolds.fasta.gz
Melap1finSC_191_AssemblyScaffolds.fasta.gz
Cunech1_AssemblyScaffolds.fasta.gz
Cirumb1_AssemblyScaffolds.fasta.gz
Phaart1_AssemblyScaffolds.fasta.gz
Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
Zyghet1_AssemblyScaffolds.fasta.gz
Melli1_AssemblyScaffolds.fasta.gz
Conco1_AssemblyScaffolds.fasta.gz
Conth1_AssemblyScaffolds.fasta.gz
Bacci1_AssemblyScaffolds.fasta.gz
Acema1_AssemblyScaffolds.fasta.gz
Chove1_AssemblyScaffolds.fasta.gz
Mellp2_3_AssemblyScaffolds.fasta.gz
Pilano1_AssemblyScaffolds.fasta.gz
Absrep1_AssemblyScaffolds.fasta.gz
Melbi2_AssemblyScaffolds.fasta.gz
Pucgr2_AssemblyScaffolds.fasta.gz
Puctr1_AssemblyScaffolds.fasta.gz
Tubme1_AssemblyScaffolds.fasta.gz
Rhior3_scaffolds.fasta.gz
Cokrec1_AssemblyScaffolds.fasta.gz
Cenge3_AssemblyScaffolds.fasta.gz
Trihyb1_AssemblyScaffolds.fasta.gz
Pucst_PST78_1_AssemblyScaffolds.fasta.gz
Croqu1_AssemblyScaffolds.fasta.gz
Denbi1_AssemblyScaffolds.fasta.gz
Rhich1_AssemblyScaffolds.fasta.gz
Leugo1_1_AssemblyScaffolds.fasta.gz
HypCO275_1_AssemblyScaffolds.fasta.gz
Ascru1_AssemblyScaffolds.fasta.gz
Mycgal1_AssemblyScaffolds.fasta.gz
Thaele1_AssemblyScaffolds.fasta.gz
Sphst1_AssemblyScaffolds.fasta.gz
Blade1_AssemblyScaffolds.fasta.gz
Rutfi1_AssemblyScaffolds.fasta.gz
Rhier1_AssemblyScaffolds.fasta.gz
Cyastr2_AssemblyScaffolds.fasta.gz
```
Copy in the files listed above:
```
while read file; do
  cp ~/fungal_genomes/JGI_1KFGs/${file} .
done < jgi_fungi_ref_genome_k21.txt
```

Add:
```
cp ~/fungal_genomes/ncbi-genome-downloads-fungi/genbank/fungi/GCA_000721785.1/GCA_000721785.1_Aureobasidium_pullulans_var._pullulans_EXF-150_assembly_version_1.0_genomic.fna.gz .
cp ~/fungal_genomes/ncbi-genome-downloads-fungi/genbank/fungi/GCA_000785685.1/GCA_000785685.1_ASM78568v1_genomic.fna.gz .
cp ~/fungal_genomes/ncbi-genome-downloads-fungi/genbank/fungi/GCA_001443065.1/GCA_001443065.1_ASM144306v1_genomic.fna.gz .
```
Then 
```
gunzip *gz
```
Edit the jgi files so that the scaffold identifiers are unique. Test on one file first, then write a loop:
```
find Neosp1_AssemblyScaffolds.fasta -type f -exec sed -i 's/scaffold/Neosp1_AssemblyScaffolds.fasta/g' {} \;

for infile in Absrep1_AssemblyScaffolds.fasta Acema1_AssemblyScaffolds.fasta Anasp1_AssemblyScaffolds.fasta Ascru1_AssemblyScaffolds.fasta Bacci1_AssemblyScaffolds.fasta Blade1_AssemblyScaffolds.fasta Cenge3_AssemblyScaffolds.fasta Chove1_AssemblyScaffolds.fasta Cirumb1_AssemblyScaffolds.fasta Cokrec1_AssemblyScaffolds.fasta Conco1_AssemblyScaffolds.fasta Conth1_AssemblyScaffolds.fasta Croqu1_AssemblyScaffolds.fasta Cunech1_AssemblyScaffolds.fasta Cyastr2_AssemblyScaffolds.fasta Denbi1_AssemblyScaffolds.fasta Gloin1_AssemblyScaffolds.fasta HypCO275_1_AssemblyScaffolds.fasta Leugo1_1_AssemblyScaffolds.fasta Melap1finSC_191_AssemblyScaffolds.fasta Melbi2_AssemblyScaffolds.fasta Melli1_AssemblyScaffolds.fasta Mellp2_3_AssemblyScaffolds.fasta Mycgal1_AssemblyScaffolds.fasta Orpsp1_1_AssemblyScaffolds.fasta Phaart1_AssemblyScaffolds.fasta Phycomyces_blakesleeanus_v2_scaffolds.fasta Pilano1_AssemblyScaffolds.fasta PirE2_1_AssemblyScaffolds.fasta Pirfi3_AssemblyScaffolds.fasta Pucgr2_AssemblyScaffolds.fasta Pucst_PST78_1_AssemblyScaffolds.fasta Puctr1_AssemblyScaffolds.fasta Rhich1_AssemblyScaffolds.fasta Rhier1_AssemblyScaffolds.fasta Rhior3_scaffolds.fasta Rutfi1_AssemblyScaffolds.fasta Sphst1_AssemblyScaffolds.fasta Thaele1_AssemblyScaffolds.fasta Trihyb1_AssemblyScaffolds.fasta Tubme1_AssemblyScaffolds.fasta Zoprh1_AssemblyScaffolds.fasta Zyghet1_AssemblyScaffolds.fasta
do 
  find $infile -type f -exec sed -i "s/scaffold/${infile}/g" {} \;
done
```

Additionally, remove "contig" and "Supercontig", as these create ambiguous duplicates
```
for infile in *a
do
  find $infile -type f -exec sed -i "s/contig/${infile}/g" {} \;
done

for infile in *a
do
  find $infile -type f -exec sed -i "s/Supercontig/${infile}/g" {} \;
done
```
Concatenate files:
```
cat *a > ref_genome_k21_fungi.fasta
```

Make a blast db:
```
makeblastdb -in ref_genome_k21_fungi.fasta -parse_seqids -dbtype nucl
```
link in olive reference genome:
```
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa .
```
Blast reference genome against the database:
```
blastn -db ref_genome_k21_fungi.fasta -query Oe6.scaffolds.fa -out Oe6.scaffolds.fa.out -outfmt 6
```

Note:
```
*** Formatting options
 -outfmt <String>
   alignment view options:
     0 = pairwise,
     1 = query-anchored showing identities,
     2 = query-anchored no identities,
     3 = flat query-anchored, show identities,
     4 = flat query-anchored, no identities,
     5 = XML Blast output,
     6 = tabular,
     7 = tabular with comment lines,
     8 = Text ASN.1,
     9 = Binary ASN.1,
    10 = Comma-separated values,
    11 = BLAST archive format (ASN.1) 
```

Note that to use ncbi `blast_formatter`, the output should be set to -outfmt 11. 

Realized that there were contigs with prefixes "LG", "c", "d", and "sc", and that "Supercontig" should be removed before contig. Edit these, and then remake blast db.

```
for infile in *a
do 
  find $infile -type f -exec sed -i "s/scaffold/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/Supercontig/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/contig/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/LG/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/sc_/${infile}_/g" {} \;
  find $infile -type f -exec sed -i "s/c_/${infile}_/g" {} \;
  find $infile -type f -exec sed -i "s/d_/${infile}_/g" {} \;
done
```

Remove the first database (ref_genome_k21) and make a new one
```
rm ref_genome_k21_fungi.fasta* Oe6.scaffolds.fa
cat *a > ref_genome_k21_fungi2.fasta
makeblastdb -in ref_genome_k21_fungi2.fasta -parse_seqids -dbtype nucl
```
Run blast again, this time with fmt 11
```
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa .
blastn -db ref_genome_k21_fungi2.fasta -query Oe6.scaffolds.fa -out Oe6.scaffolds.fa.2.out.asn -outfmt 11
```

Output as different formats
```
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 5 -out Oe6.scaffolds.fa.2.out.xml
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 6 -out Oe6.scaffolds.fa.2.out.tab
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 0 -out Oe6.scaffolds.fa.2.out.pw
```
And in an effort to reduce data for better visualization options, format to show only the best hit
```
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 5 -max_target_seqs 1 -out Oe6.scaffolds.fa.2.max1.out.xml
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 6 -max_target_seqs 1 -out Oe6.scaffolds.fa.2.max1.out.tab
blast_formatter -archive Oe6.scaffolds.fa.2.out.asn -outfmt 0 -max_target_seqs 1 -out Oe6.scaffolds.fa.2.max1.out.pw
```
split blast pw

### Remember to rename `blast_pw_split` to `blast_pw_split_max1` 
```
mkdir blast_pw_split
cd blast_pw_split/
ln -s ../Oe6.scaffolds.fa.2.max1.out.pw .
awk '/Query= Oe6_s/{n++}{print > n ".pw" }' Oe6.scaffolds.fa.2.max1.out.pw
```
## Visualizing the blast output
```
# install circos and circletto dependencies
sudo apt-get install bioperl
sudo apt-get -y install libgd-perl
cd

# install circos
mkdir circos
cd circos
curl -O http://dib-training.ucdavis.edu.s3.amazonaws.com/metagenomics-scripps-2016-10-12/circos-0.69-3.tar.gz
tar -xvzf circos-0.69-3.tar.gz

export PATH=~/circos/circos-0.69-3/bin:$PATH
```
Install circos modules
```
circos -modules > modules
grep missing modules |cut -f13 -d " " > missing_modules
for mod in $(cat missing_modules);
  do
  sudo cpan install $mod;
  done
```
Install circos tools:
```
cd ~/circos
curl -LO http://circos.ca/distribution/circos-tools-0.22.tgz
tar -xvzf circos-tools-0.22.tgz
```
Run circoletto
changed `max_ideograms` in `~/circos/circos-0.69-3/etc/housekeeping.conf` to 20000.  :)
Also, note that the commands below are running on identity, with blue indicating 50% or above, green indicating 75% or above, and orange 99.9999% or above.
```
# note absolute paths were added to the circoletto.py code first.
~/Circoletto/circoletto.pl --cpus 4 --blastout 3746.pw --out_type png --w_hits --score2colour id --scoreratio2colour max --abscolour --maxB1 50 --maxG2 75 --maxO3 99.9999 --out_size 750 --annocolour scores --ribocolours2allow '(blue|green|orange|red)' --z_by score 
```

The above command worked, now run on all of the Oe6 scaffolds (in this folder, the pw files only have the best hits. Do this first, then repeat but with the complete pw file, not just with max1.
```
rm Oe6.scaffolds.fa.2.max1.out.pw 
for i in *pw 
do
  ~/Circoletto/circoletto.pl --cpus 4 --blastout $i --out_type png --w_hits --score2colour id --scoreratio2colour max --abscolour --maxB1 50 --maxG2 75 --maxO3 99.9999 --out_size 750 --annocolour scores --ribocolours2allow '(blue|green|orange|red)' --z_by score 
done
```

Repeat. First, set up files to run it on
```
cd /mnt/work/blast/sbt_fungi_blastdb/
mkdir blast_pw_split_all
cd blast_pw_split_all
ln -s ../Oe6.scaffolds.fa.2.out.pw .
awk '/Query= Oe6_s/{n++}{print > n ".pw" }' Oe6.scaffolds.fa.2.out.pw
rm Oe6.scaffolds.fa.2.out.pw
```

Run circoletto again
```
for i in *pw 
do
  ~/Circoletto/circoletto.pl --cpus 4 --blastout $i --out_type png --w_hits --score2colour id --scoreratio2colour max --abscolour --maxB1 50 --maxG2 75 --maxO3 99.9999 --out_size 750 --annocolour scores --ribocolours2allow '(blue|green|orange|red)' --z_by score 
done
```
