Calculate the genome signature of each of the scaffolds in Oe6. See if any scaffolds differ, and if so, if these are the same scaffolds that were identified by BLAST as having fungal content.

calculate di, tri, tetra, and penta nucleotide frequencies of olive scaffolds. Use scaled 10kbp; resolution not necessary.
```
cd /mnt/work/ref_genome
mkdir scaffolds
cd scaffolds
ln -s ../Oe6.scaffolds.fa .

awk '/>Oe6_s/{n++}{print > n ".scaffold.fa" }' Oe6.scaffolds.fa

for infile in *scaffold.fa
do
  sourmash compute -k 2,3,4,5 --scaled 5 --track-abundance -o ${infile}.s5.k2-5.sig ${infile}
done
```

Sigs keep crashing jupyter notebook. Remove fasta files from directory
```
mkdir fastas
for infile in *.fa
do
  mv $infile ./fastas/$infile
done
```

Still crashed, run compare on the command line
```
sourmash compare -k 4 -o Oe6_scaffolds_k4.comp *sig
sourmash compare -k 2 -o Oe6_scaffolds_k2.comp *sig
sourmash compare -k 3 -o Oe6_scaffolds_k3.comp *sig
sourmash compare -k 5 -o Oe6_scaffolds_k5.comp *sig
```


Other tools that can be used: SigHunt, BLSOM ProDeGe
