Quantify read counts with Salmon

Make sure to activate `pondenv`, which was created during the eelpond protocol.
```
source ~/pondenv/bin/activate
```

```
cd /mnt/work/eelpond/
mkdir -p quant
cd quant
ln -s /mnt/work/eelpond/assembly/trinity_out_dir/Trinity.fasta .
```

Index transcriptome
```
salmon index --index olea-fruit --transcripts Trinity.fasta --type quasi
```

Link to qc'd reads
```
ln -s ../../quality/*qc.fq.gz .
ls
```  

Run Salmon for single end data
```
for file in SRR*
  do
    salmon quant -i olea-fruit -p 2 -l IU -r <(gunzip -c ${file}) -o ${file}quant
done
```

Run Salmon for paired-end data

**NB re-check naming scheme before running this**
```
for R1 in *R1*.qc.fq.gz
do
  sample=$(basename $R1 extract.qc.fq.gz)
  echo sample is $sample, R1 is $R1
  R2=${R1/R1/R2}
  echo R2 is $R2
  salmon quant -i olea-fruit -p 2 -l IU -1 <(gunzip -c $R1) -2 <(gunzip -c $R2) -o ${sample}quant
done
```
