Use HISAT to map reads back to Trinity transcriptome

Install hisat
```
mkdir $HOME/hisat_bin
export PATH=$HOME/hisat_bin:$PATH
```
```
wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.0.5-Linux_x86_64.zip
unzip hisat2-2.0.5-Linux_x86_64.zip
cp hisat2-2.0.5/hisat* hisat2-2.0.5/*.py $HOME/hisat_bin
```
If samtools is not installed, install it
```
wget https://github.com/samtools/samtools/releases/download/1.3.1/samtools-1.3.1.tar.bz2
tar jxvf samtools-1.3.1.tar.bz2
cd samtools-1.3.1
make
cd ..
cp samtools-1.3.1/samtools $HOME/hisat_bin
```

Prep for indexing. Use dammit gtf file.
```
cd /mnt/work/eelpond/
mkdir align
cd align
$HOME/hisat_bin/extract_splice_sites.py /mnt/work/eelpond/annotation/*.gtf > olea-fruit.ss 
$HOME/hisat_bin/extract_exons.py /mnt/work/eelpond/annotation/*.gtf > olea-fruit.exon
```

Build index
```
mkdir index
cd index
$HOME/hisat_bin/hisat2-build --ss ../olea-fruit.ss --exon ../olea-fruit.exon /mnt/work/eelpond/assembly/trinity_out_dir/Trinity.fasta/ olea-fruit-index
```

Align single-end reads
```
cd /mnt/work/eelpond/align
ln -s /mnt/work/single/quality/*qc.fq.gz .
ls

for f in SRR* 
  do
    $HOME/hisat_bin/hisat2 -p 3 -x ./index/olea-fruit-index -U ${f}  -S ${f}.sam --met-file ${f}.met --un-gz unaligned_${f}.gz
done
```
Align paired end reads
```
cd /mnt/work/eelpond/align
$HOME/hisat_bin/hisat2 -p 3 -x ./index/olea-fruit-index -1 FIRST_READ -2 SECOND READ -S ERR...sam --met-file ERR...met --un-gz unaligned-se_ERR...gz --un-conc-gz unaligned-pe_ERR...gz
```

Convert sam to bam

Link in transcriptome
```
ln -s /mnt/work/eelpond/assembly/trinity_out_dir/Trinity.fasta
ls
```
Index transcriptome, convert to bam, sort bam, and index the sorted bam. 
```
for i in *.sam
do
  samtools faidx Trinity.fasta
  samtools view -bt Trinity.fasta.fai ${i} > ${i}.bam
  samtools sort ${i}.bam -o ${i}.sorted.bam
  samtools index ${i}.sorted.bam
  
done
```

Delete sam files, intermediate bam files. 
  

