Align to reference genome. Write reads out that don't align. Run Trinity on these these reads. Annotate these for fungi and bacteria. 
Goal to downsize number of transcripts, and hopefully isolate some of the microbes from the olives. 

Run on qc'd reads from eelpond protocol (see protocol in eelpond directory)

Performed indexing on a m4.10xlarge (160 gb ram, 40 cpu)

*NB* I upgraded the instance that I had been using for the eelpond protocol. If software installation is missing, this is why.

make directory to store executables
```
mkdir $HOME/hisat_bin
export PATH=$HOME/hisat_bin:$PATH
```

Install HISAT2 
```
wget ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.0.5-Linux_x86_64.zip
unzip hisat2-2.0.5-Linux_x86_64.zip
cp hisat2-2.0.5/hisat* hisat2-2.0.5/*.py $HOME/hisat_bin
```

Install samtools
```
wget https://github.com/samtools/samtools/releases/download/1.3.1/samtools-1.3.1.tar.bz2
tar jxvf samtools-1.3.1.tar.bz2
cd samtools-1.3.1
make
cd ..
cp samtools-1.3.1/samtools $HOME/hisat_bin
```

Get olive genome

First tried `wget ftp://climb.genomics.cn/pub/10.5524/100001_101000/100201/Oe6.scaffolds.fa.gz` however this link was broken and downloaded a corrupted file. 

```
cd /mnt/work/
mkdir hisat
cd hisat
mkdir genome
cd genome
wget http://denovo.cnag.cat/genomes/olive/download/Oe6/Oe6.scaffolds.fa.gz
```

Get olive annotation file
```
cd /mnt/work/hisat
mkdir genes
cd genes
wget ftp://climb.genomics.cn/pub/10.5524/100001_101000/100201/OE6A.func.desc.gff3.gz
```

try another annotation file as well
```
wget ftp://climb.genomics.cn/pub/10.5524/100001_101000/100201/oe6_proteins.gff3.gz
gunzip oe6_proteins.gff3.gz
```
And try
```
wget ftp://climb.genomics.cn/pub/10.5524/100001_101000/100201/oe6_geneid.gff3
```

Get program to convert gff to gtf
```
cd ~
wget http://genometools.org/pub/genometools-1.5.9.tar.gz
tar -xvf genometools-1.5.9.tar.gz
cd genometools-1.5.9
make cairo=no
sudo make cairo=no install
cd ..
cp genometools-1.5.9/scripts/* $HOME/hisat_bin
```

convert gff to gtf
```
oe6_geneid.gff3
gt gff3_to_gtf /mnt/work/hisat/genes/oe6_geneid.gff3 > /mnt/work/hisat/genes/oe6_geneid.gtf
```
###### Note to self: some of these files may be in /mnt/work/star/genes

Prep for indexing
```
$HOME/hisat_bin/extract_splice_sites.py /mnt/work/hisat/genes/oe6_geneid.gtf > olive.ss 
$HOME/hisat_bin/extract_exons.py /mnt/work/hisat/genes/oe6_geneid.gtf > olive.exon
```

update instance to an r4.16xlarge:	cpu 64 ecu 195 ram 488 GB. If it breaks, I quit.

index genome
```
$HOME/hisat_bin/hisat2-build --ss olive.ss --exon olive.exon /mnt/work/hisat/genome/Oe6.scaffolds.fa olive_index
```
```
cd /mnt/work/hisat
mkdir index
for infile in *ht2
  do
    mv $infile /mnt/work/hisat/index/${infile}
done
```

Prepare directories for the hisat output files
```
mkdir alignments
mkdir unaligned
```

Run alignments with output of unaligned fastq files.
```
cd /mnt/work/quality/
for f in *qc.fq 
  do
    $HOME/hisat_bin/hisat2 -p 3 -x /mnt/work/hisat/index/olive_index -U ${f}  -S /mnt/work/hisat/alignments/${f}.sam --met-file /mnt/work/hisat/alignments/${f}.met --un-gz /mnt/work/hisat/unaligned/unaligned_${f}.gz
done
```

Ran out of space, add another 100 gb to instance
```
cd /mnt/work/quality/
```
Re-run on last file that terminated because of space problems
```
$HOME/hisat_bin/hisat2 -p 3 -x /mnt/work/hisat/index/olive_index -U SRR926287qc.fq -S /mnt/work/hisat/alignments/SRR926287qc.fq.sam --met-file /mnt/work/hisat/alignments/SRR926287qc.fq.met --un-gz /mnt/work/hisat/unaligned/unaligned_SRR926287qc.fq.gz
```

Note this instance had a python2 and python3 conflict because of poor installation instructions in eelpond protocol (or, my poor ability to follow them...). Instance was deleted and software was reinstalled, but some software, such as hisat, was left out. Reinstall as above if needed in the future. 

Convert sam to bam format to save space
```
for infile in *.sam
  do
    samtools view -bS ${infile} > ${infile}.bam
done
```


