Update sourmash from github. Delete all signatures, and rerun. 

Run `sourmash sbt_gather` on the quality controlled olive fruit reads. 

Here, quality controlled is defined as trimmomatic and khmer `trim-low-abund --variable coverage <...>` (see 0.preprocessing).

```
cd /mnt/work/sourmash
mkdir sbt_gather
cd sbt_gather
ln -s ../signatures/*sig .
```

```
source ~/sourmashEnv2/bin/activate
for infile in *.sig
  do
    sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json --threshold=0.001 -o fungal_k31_${infile}.txt ${infile}
 done
```

Run sourmash again...the first few files were empty and I'm skeptical of this

4/25/17: there was a bug preventing the output from writing to files appropriately.
```
for infile in *.sig
  do
    sourmash sbt_gather -k 31 ~/sourmash_SBTs/fungal_4.13.17.sbt.json --threshold=0.001 -o fungal_trial2_k31_${infile}.txt ${infile}
 done
 ```
 
 Check for reference genome content
```
cd /mnt/work/sourmash/sbt_gather
source ~/sourmashEnv2/bin/activate
for infile in *.sig
  do
    sourmash sbt_gather -k 31 ~/sourmash_SBTs/olive_ref/olive_ref_4.18.17.sbt.json --threshold=0.001 -o olive_k31_${infile}.txt ${infile}
 done
```

Run against the k31 microbes sbt to see what is present other than fungus
Need to re-run this. 
```
for infile in *.sig
  do
    sourmash sbt_gather -k 31 ~/sourmash_SBTs/microbes.k31.sbt.json --threshold=0.001 -o microbe_k31_${infile}.txt ${infile}
done
```

-----------------------------------
 
 4/27/17 after updating sourmash to the metagenomes-2017 branch, and after re-calculating signatures at a `--scaled 1000` for all RNA-seq files and for the fungal SBT, re-run these `sbt_gather` on RNA-seq samples.
 
 First, move all of the above runs in to a folder named `sig`, then make the new `sbt_gather` output write to a folder named `sig3`
 ```
 cd /mnt/work/sourmash/sbt_gather
 mkdir sig
 
 for infile in *
 do 
   mv $infile ./sig/$infile
 done
 
 mkdir sig3
 cd sig3
 ```
 
Link in signatures files from folder `signature3`, which contain signatures from RNA-seq reads trimmed with `trim-low-abund.py -v`, and with signatures calculated at `-k 21,31,51` and `--scaled 1000`
```
ln -s ../../signatures3/*sig3 .
```
 
Use `sbt_gather` with the new SBTs where all fungal sigs were calculated with `--scaled 1000` and `-k 21,31,51`.
```
cd /mnt/work/sourmash/sbt_gather/*sig3

for infile in *sig3 
do
    /usr/bin/time -v -o time_fungi_k21_s1000.txt -- sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json $infile --threshold-bp=0 -o fungi_k21__s1000_{infile}.txt
done
```
Because this takes so long, write individual commands for different files and execute in parallel in different tmux windows. Also make the the threshold bp 1000. 

RUNNING:
```
sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR1573503qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR1573503qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR1574328qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR1574328qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR1574719qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR1574719qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR1574772qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR1574772qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR4236909qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR4236909qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR4236911qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR4236911qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR4236910qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR4236910qc.abundtrim.fq.gz.sig3.txt

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s1000/fungal_k21_s1000.sbt.json SRR4236912qc.abundtrim.fq.gz.sig3 --threshold-bp=1000 -o fungi_k21_s1000_SRR4236912qc.abundtrim.fq.gz.sig3.txt
```

-------------------------------------------
5/15/17

Andddd never finished running. The scaled value was too low and the threshold BP was too low. Re-run sourmash with a higher threshold bp (100k) and with the 10k fungal index.


```
sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json ERR1346597.pe+orphans.qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_ERR1346597.pe+orphans.qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1573503qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR1573503qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574328qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR1574328qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574719qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR1574719qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574772qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR1574772qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236909qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR4236909qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236911qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR4236911qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236910qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR4236910qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236912qc.abundtrim.fq.gz.sig3 --threshold-bp=100000 -o fungi_k21_s10k_SRR4236912qc.abundtrim.fq.gz.sig3.csv
```
 
 
 And because that was so quick, run again with 50k threshold bp. 
 ```
sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json ERR1346597.pe+orphans.qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_ERR1346597.pe+orphans.qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1573503qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR1573503qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574328qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR1574328qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574719qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR1574719qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR1574772qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR1574772qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236909qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR4236909qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236911qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR4236911qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236910qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR4236910qc.abundtrim.fq.gz.sig3.csv

sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json SRR4236912qc.abundtrim.fq.gz.sig3 --threshold-bp=50000 -o fungi_k21_s10k_tbp50k_SRR4236912qc.abundtrim.fq.gz.sig3.csv
```
-------------------------------------------
5/17/16
The above commands worked fine, but there was an oddity in the output. Re-run after updating the sourmash installation to the master branch.

First, move files
```
mkdir 2017-metagenome-branch_output
for infile in *csv
do
  mv $infile ./2017-metagenome-branch_output/$infile
done
```

Re-run fungi, this time as a loop because it is fairly fast. Set threshold to 10kbp.
```
for infile in *sig3
do
  sourmash sbt_gather -k 21 ~/sourmash_SBTs/fungal_k21_s10k/fungal_k21_s10k.sbt.json --threshold-bp=10000 -o fungi_k21_s10k_tbp10k_${infile}.csv $infile
done
```

Run with genbank as sbt
```
for infile in *sig3
do
  sourmash sbt_gather -k 21 ~/sourmash_SBTs/genbank_k21_17-5-9/genbank-k21.sbt.json --threshold-bp=10000 -o genbank_k21_s10k_tbp10k_${infile}.csv $infile
done
```


Run with refseq as sbt
```
for infile in *sig3
do
  sourmash sbt_gather -k 21 ~/sourmash_SBTs/refseq_k21_17-5-9/refseq-k21.sbt.json --threshold-bp=10000 -o refseq_k21_s10k_tbp10k_${infile}.csv $infile
done
```


Run with new index fungal sigs 10k k 31
```
for infile in *sig3
do
  sourmash gather -k 31 --threshold-bp=10000 -o fungi_k31_s10k_tbp10k_${infile}.csv --save-matches fungi_k31_s10k_tbp10k_${infile}.csv ${infile} ~/sourmash_SBTs/fungal_k31_s10k/fungal_k31_s10k.sbt.json
done
```
