I got very whacky results from sbt_gather when only one signature was in the SBT. Use sourmash compare to estimate the similarity between the olive reference genome and olive fruit RNA-seq reads.
###### Note that these are preliminary results that will likely be off by 1-2% points, as there was a small bug in the code and the version of sourmash was not updated before running this. 

Link in the reference genome to the signatures folder
```
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa.sig .
```

Link in the signature reads to the compare folder
```
cd /mnt/work/sourmash
mkdir compare
cd compare
ln -s ../signatures/*sig .
```

Activate the environment and run sourmash
```
source sourmashEnv2/bin/activate
sourmash compare --ignore-abundance ./*sig
```

Returned problem with max hash, where max hash of Oe6 was 36893488147419104 and max hash of ERR\* was 1844674407370955. This is annoying. 

Recalculate signatures with abundances ignored. 

Link in fasta and fastq files
```
mkdir /mnt/work/sourmash/signatures2
cd signatures2
ln -s /mnt/work/sourmash/trim-low-abund/*abundtrim.gz .
ln -s /mnt/work/ref_genome/Oe6.scaffolds.fa.gz .
```

recalculate signatures
```
for infile in *gz
do
  sourmash compute -k 31 --scaled 10000 -o ${infile}.sig2 ${infile}
done
```

Activate the environment and run sourmash
```
source sourmashEnv2/bin/activate
sourmash compare --ignore-abundance ./*sig2
```
