Because the megahit assembler was run on transcripts, and because the samples contain both fungal and bacterial reads, try using Prokka and see how the results turn out.

Install Prokka
```
cd ~/
wget http://www.vicbioinformatics.com/prokka-1.11.tar.gz
tar -xvzf prokka-1.11.tar.gz
```

Install dependencies
```
sudo apt-get install bioperl libdatetime-perl libxml-simple-perl libdigest-md5-perl
sudo perl -MCPAN -e shell
sudo perl -MCPAN -e 'install "XML::Simple"'
```
```
export PATH=$PATH:$HOME/prokka-1.11/bin
prokka --setupdb
```

Organize and link in files
```
cd /mnt/work/hisat/unaligned/
mkdir prokka
cd prokka
ln -s ../megahit/*megahit* .
```

Run prokka
```
# --centre and --locustag are a work around. See https://github.com/tseemann/prokka/issues/135
for infile in ./*/final.contigs.fa
    do
    j=$(dirname $infile)
    k=$(basename $j)
    prokka ${infile} --metagenome --mincontiglen 200 --cpus 3 --centre C --locustag L --outdir ${k}_prokka
done
```    
Note these important defaults:
```
        Defaults:
         --metagenome      Improve gene predictions for highly fragmented genomes (default OFF)
         --evalue [n.n]    Similarity e-value cut-off (default '1e-06')
```
