 Run salmon
 
 Oraganize files and set things up. Link in all reads and use `--libType A` for salmon to automatically infer what library type is present.
 ```
 cd /mnt/work/hisat/unaligned/
 mkdir salmon
 cd salmon
 ln -s ../*gz .
 ```
 Index the transcriptomes with salmon
 ```
 ln -s ../megahit/*megahit* .
 for infile in ./*/final.contigs.fa;
    do 
    j=$(dirname $infile)
    k=$(basename $j)
    salmon index --index ${k}_index --type quasi -k 31 --transcripts ${infile}
 done
 ```
 Run salmon
 ```
 for infile in *.fq.gz
 do
     salmon quant -i ${infile}.megahit_index --libType A -r ${infile} -o ${infile}.quant --seqBias --gcBias
 done
```

Run salmon on the lone little sample that didn't fit the loop above
```
 salmon quant -i megahit_ERR1346597_index --libType A -1 unaligned-pe_ERR1346597_1.qc.fq.gz  -2 unaligned-pe_ERR1346597_2.qc.fq.gz -o unaligned-pe_ERR1346597.quant --seqBias --gcBias
```

Parse count data
```
 curl -L -O https://github.com/ngs-docs/2016-aug-nonmodel-rnaseq/raw/master/files/gather-counts.py
 python2 gather-counts.py
 ```
