FungiDB

Search for all transcripts in FungiDB
Exclude all transcripts that occur in non-fungi eukaryotes, bacteria, and archaea
Download sequences (1.98 GB)

```
cd /mnt/work/ref_genome
mkdir fungiDB
cd fungiDB
gunzip boolean_question_TranscriptRecordClasses_TranscriptRecordClass.fasta.gz
```
Blast against Oe6 scaffolds with a low e-value threshold
```
blastn -db ../Oe6.scaffolds.fa -query boolean_question_TranscriptRecordClasses_TranscriptRecordClass.fasta -out fungidb_blastn_vs_Oe6_scaffolds.fa.out.asn -evalue .000001 -outfmt 11
blast_formatter -archive fungidb_blastn_vs_Oe6_scaffolds.fa.out.asn -outfmt 6 -out fungidb_blastn_vs_Oe6_scaffolds.fa.out.tab
blast_formatter -archive fungidb_blastn_vs_Oe6_scaffolds.fa.out.asn -outfmt 0 -out fungidb_blastn_vs_Oe6_scaffolds.fa.out.pw

# max target seqs 10
blast_formatter -archive fungidb_blastn_vs_Oe6_scaffolds.fa.out.asn -outfmt 6 -max_target_seqs 10 -out fungidb_blastn_vs_Oe6_scaffolds.fa.max1.out.tab
blast_formatter -archive fungidb_blastn_vs_Oe6_scaffolds.fa.out.asn -outfmt 0 -max_target_seqs 10 -out fungidb_blastn_vs_Oe6_scaffolds.fa.max1.out.pw
```

Many of the blast returns appeared as though they may have a significant degree of homology with plants. Try EggNOG FuNOG database instead. 

*NB* The code below was not successful yet
```
wget http://eggnogdb.embl.de/download/eggnog_4.5/data/fuNOG/fuNOG.trimmed_algs.tar.gz
tar xf fuNOG.trimmed_algs.tar.gz
cd fuNOG_trimmed_algs/
cat *fa > fuNOG.ALL.meta_trimmed.fa

# lots of errors, but that's because there are a lot of dashes et al. and maybe other things --- Run again line 22000 a problem?

blastn -db ../Oe6.scaffolds.fa -query ./fuNOG_trimmed_algs/fuNOG.ALL.meta_trimmed.fa -out fuNOG_trimmed_algs.asn -evalue .000001 -outfmt 11
# blast_formatter -archive fuNOG_trimmed_algs.asn -outfmt 6 -out fuNOG_trimmed_algs.tab
# blast_formatter -archive fuNOG_trimmed_algs.asn -outfmt 0 -out fuNOG_trimmed_algs.asn.pw

# Max target seqs
# blast_formatter -archive fuNOG_trimmed_algs.asn -max_target_seqs 10 -outfmt 6 -out fuNOG_trimmed_algs.tab 
# blast_formatter -archive fuNOG_trimmed_algs.asn -max_target_seqs 10 -outfmt 0 -out fuNOG_trimmed_algs.asn.pw
```

Some of these things seem promising. Filter out candidates by blasting against other reference genomes. Those that have hits, exclude from blast against olive reference genome.

Download genomes for arabidopsis, physcomitrens, oryza sativa, solanum lycopersicum, and populus trichocarpa.
```
cd /mnt/work/other_ref_genomes
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-35/fasta/solanum_lycopersicum/dna/Solanum_lycopersicum.SL2.50.dna.toplevel.fa.gz
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-35/fasta/populus_trichocarpa/dna/Populus_trichocarpa.JGI2.0.dna.toplevel.fa.gz
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-35/fasta/oryza_sativa/dna/Oryza_sativa.IRGSP-1.0.dna.toplevel.fa.gz
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-35/fasta/physcomitrella_patens/dna/Physcomitrella_patens.ASM242v1.dna.toplevel.fa.gz
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-35/fasta/arabidopsis_thaliana/dna/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz

gunzip .gz
```

Place all in one file
```
cat *fa > other_ref_genomes.fa
```

Make a blast database
```
cd /mnt/work/ref_genome/fungiDB/
ln -s ../../other_ref_genomes/other_ref_genomes.fa .
makeblastdb -in other_ref_genomes.fa -dbtype nucl
```

Blast fungiDB against database
```
blastn -db other_ref_genomes.fa -query boolean_question_TranscriptRecordClasses_TranscriptRecordClass.fasta -out fungidb_blastn_vs_other_ref_genomes.fa.out.asn -evalue .0000000001 -outfmt 11

blast_formatter -archive fungidb_blastn_vs_other_ref_genomes.fa.out.asn -outfmt 6 -out fungidb_blastn_vs_other_ref_genomes.fa.out.tab 
```

Grep for ">" in the fungiDB file. Output to a new txt file. Import this in to R. Import the blast output tab file in to R. Remove files from the grep vector that occur in the blast tab dataframe. Out put this as a new text file. Use bioawk to grep for these from the fungiDB file. Blast these against Oe6.

Grep for fasta names:
```
grep ">" boolean_question_TranscriptRecordClasses_TranscriptRecordClass.fasta > boolean_question_TranscriptRecordClasses_TranscriptRecordClass_fastaNames.txt
```
Things in R
```
# Read in all fungi db names and parse
fungidb_names<-read.csv("boolean_question_TranscriptRecordClasses_TranscriptRecordClass_fastaNames.txt", header = FALSE, sep = "|")
fungidb_names<-fungidb_names[,1]
fungidb_names<-gsub( ">", "", as.character(fungidb_names))
fungidb_names<-gsub( " ", "", as.character(fungidb_names))

# Read in blast tab output and parse
other_genomes_hits<-read.table("fungidb_blastn_vs_other_ref_genomes.fa.out.tab", header = FALSE)
other_genomes_hits<-other_genomes_hits[,1]
other_genomes_hits<-as.character(other_genomes_hits)

# get the difference
setdiff(fungidb_names, other_genomes_hits)
not_in_other_plants<-setdiff(fungidb_names, other_genomes_hits)

# check that it worked
length(fungidb_names)
length(other_genomes_hits)
length(not_in_other_plants)

# Yay data reduction!
write.table(not_in_other_plants, "fungidb_names_not_in_other_plants.txt", col.names=FALSE, row.names=FALSE, quote = FALSE)
```

```
git clone 
while read inline
do
 cat fasta.fa | bioawk -c fastx ' $name ~ /^A/ { print ">" $name "\n" $seq }'
```

