Align all of the genomes identified by sourmash sbt_gather at k 31 to the olive reference genome, and see if large chunks align or if small pieces do. 

Install MUMmer, which contains nucmer. 
```
cd ~
wget https://github.com/gmarcais/mummer/releases/download/v4.0.0beta/mummer-4.0.0beta.tar.gz
tar xf mummer-4.0.0beta.tar.gz 
cd mummer-4.0.0beta
./configure 
make
make install
```

Link in files that match at k31 from sourmash run.

Genbank and RefSeq genomes:
```
MSDY01000045.1 Aureobasidium sp. FSWF8-4 FSW8-bin-4-scaffold_100, whole genome shotgun sequence
NW_017306190.1 Alternaria alternata strain SRC1lrK2f unplaced genomic scaffold CC77scaffold_1, whole genome shotgun sequence
MKXH02000317.1 Puccinia striiformis f. sp. tritici strain 38S102 Scaffold_1, whole genome shotgun sequence
LZZZ01000001.1 Ustilago cynodontis strain NBRC 9727 NODE_1, whole genome shotgun sequence
LVWM01000001.1 Aureobasidium pullulans isolate Santander contig_1, whole genome shotgun sequence
KE148146.1 Ophiostoma piceae UAMH 11346 chromosome Unknown scf01, whole genome shotgun sequence
KN360965.1 Uromyces viciae-fabae strain I2 unplaced genomic scaffold scaffold_1, whole genome shotgun sequence
KN715891.1 Penicillium chrysogenum KF-25 unplaced genomic scaffold scaffold1, whole genome shotgun sequence
KQ435302.1 Mucor circinelloides strain WJ11 unplaced genomic scaffold scaffold00001, whole genome shotgun sequence
KQ503367.1 Puccinia psidii isolate Aus_3 unplaced genomic scaffold scaffold_1, whole genome shotgun sequence
KV453841.1 Tortispora caseinolytica NRRL Y-17796 unplaced genomic scaffold CANCAscaffold_1, whole genome shotgun sequence
LADD01000001.1 Puccinia triticina Race 77 isolate 77-9 contig00001, whole genome shotgun sequence
LMTL01000100.1 Candida sojae strain GF41 100_contig, whole genome shotgun sequence
JEMT01000001.1 Rhizophagus irregularis DAOM 197198w jcf7180003165201, whole genome shotgun sequence
JNEG01000001.1 Cunninghamella bertholletiae 175 ctg7180000053040_1, whole genome shotgun sequence
JNHX01000001.1 Amanita muscaria var. guessowii Koide BX008 seq1, whole genome shotgun sequence
AABY01000832.1 Saccharomyces paradoxus NRRL Y-17217 contig_531, whole genome shotgun sequence
ABPC01006145.1 Saccharomyces cerevisiae M22 Contig3380, whole genome shotgun sequence
ADAS02000001.1 Puccinia triticina 1-1 BBBD Race 1 supercont2.1, whole genome shotgun sequence
AEIW01000001.1 Cordyceps cicadae BA-001 contig00001, whole genome shotgun sequence
AFRD01000864.1 Periglandula ipomoeae IasaF13 contig27004, whole genome shotgun sequence
AZRP01000001.1 Puccinia triticina Race 106 contig00001, whole genome shotgun sequence
CAGI01000713.1 Ustilago hordei Uh4857-4 WGS project CAGI00000000 data, contig UHOR_scaffold_4.01268, whole genome shotgun sequence
CM001110.1 Saccharomyces cerevisiae FostersB chromosome I, whole genome shotgun sequence
CVRD01000001.1 Clavaria fumosa genome assembly Cla_fum_l100.fa, contig 40, whole genome shotgun sequence
DS268403.1 Coccidioides immitis RMSCC 3703 supercont1.286 genomic scaffold, whole genome shotgun sequence
FMVU01002546.1 Ustilago bromivora strain UB2 genome assembly, contig: C10459, whole genome shotgun sequence
```

Create a files with the following JGI samples in it:
```
Absrep1_AssemblyScaffolds.fasta.gz
Acain1_AssemblyScaffolds.fasta.gz
Acema1_AssemblyScaffolds.fasta.gz
Amath1_AssemblyScaffolds.fasta.gz
Amore1_AssemblyScaffolds.fasta.gz
Anasp1_AssemblyScaffolds.fasta.gz
Armga1_AssemblyScaffolds.fasta.gz
Ascni1_AssemblyScaffolds.fasta.gz
Aspbru1_AssemblyScaffolds.fasta.gz
Aspca3_scaffolds.fasta.gz
Aspell1_AssemblyScaffolds.fasta.gz
Aspfij1_AssemblyScaffolds.fasta.gz
Aspfo1_AssemblyScaffolds.fasta.gz
Aspfu_A1163_1_AssemblyScaffolds.fasta.gz
Aspgl1_AssemblyScaffolds.fasta.gz
Asphet1_AssemblyScaffolds.fasta.gz
Aspibe1_AssemblyScaffolds.fasta.gz
Aspind1_AssemblyScaffolds.fasta.gz
Aspph1_AssemblyScaffolds.fasta.gz
Aspvio1_AssemblyScaffolds.fasta.gz
Aurpu_var_pul1_AssemblyScaffolds.fasta.gz
Aurpu_var_sub1_AssemblyScaffolds.fasta.gz
Basme2finSC_AssemblyScaffolds.fasta.gz
Bimnz1_AssemblyScaffolds.fasta.gz
Bisnum1_AssemblyScaffolds.fasta.gz
Blade1_AssemblyScaffolds.fasta.gz
Blugr1_AssemblyScaffolds.fasta.gz
Botci1_AssemblyScaffolds.fasta.gz
Bulin1_AssemblyScaffolds.fasta.gz
Calful1_AssemblyScaffolds.fasta.gz
Canar1_AssemblyScaffolds.fasta.gz
Chove1_AssemblyScaffolds.fasta.gz
Cirumb1_AssemblyScaffolds.fasta.gz
Clabul1_AssemblyScaffolds.fasta.gz
Claca1_AssemblyScaffolds.fasta.gz
Cloaq1_AssemblyScaffolds.fasta.gz
CocheC5_3_AssemblyScaffolds.fasta.gz
Cocim1_AssemblyScaffolds.fasta.gz
Coclu2_AssemblyScaffolds.fasta.gz
Cokrec1_AssemblyScaffolds.fasta.gz
Colca1_AssemblyScaffolds.fasta.gz
Colce1_AssemblyScaffolds.fasta.gz
Colfi1_AssemblyScaffolds.fasta.gz
Colgr1_AssemblyScaffolds.fasta.gz
Colhig2_AssemblyScaffolds.fasta.gz
Collu1_AssemblyScaffolds.fasta.gz
Colph1_AssemblyScaffolds.fasta.gz
Colso1_AssemblyScaffolds.fasta.gz
Colzo1_AssemblyScaffolds.fasta.gz
Conlig1_AssemblyScaffolds.fasta.gz
Conth1_AssemblyScaffolds.fasta.gz
Corca1_AssemblyScaffolds.fasta.gz
Corma2_AssemblyScaffolds.fasta.gz
Crevar1_AssemblyScaffolds.fasta.gz
Croqu1_AssemblyScaffolds.fasta.gz
Crypo1_AssemblyScaffolds.fasta.gz
Cunech1_AssemblyScaffolds.fasta.gz
DalEC12_1_AssemblyScaffolds.fasta.gz
Delst1_AssemblyScaffolds.fasta.gz
Denbi1_AssemblyScaffolds.fasta.gz
Eurhe1_AssemblyScaffolds.fasta.gz
Fenlin1_AssemblyScaffolds.fasta.gz
Gaumor1_1_AssemblyScaffolds.fasta.gz
Gilper1_AssemblyScaffolds.fasta.gz
Gloin1_AssemblyScaffolds.fasta.gz
Gonbut1_AssemblyScaffolds.fasta.gz
Hanva1_1_AssemblyScaffolds.fasta.gz
Hetpy1_AssemblyScaffolds.fasta.gz
Hymvar1_AssemblyScaffolds.fasta.gz
Kalpfe1_AssemblyScaffolds.fasta.gz
Lobtra1_AssemblyScaffolds.fasta.gz
Lodelo1_AssemblyScaffolds.fasta.gz
Lycper1_AssemblyScaffolds.fasta.gz
Macph1_AssemblyScaffolds.fasta.gz
Maggr1_AssemblyScaffolds.fasta.gz
Magpo1_AssemblyScaffolds.fasta.gz
Marbr1_AssemblyScaffolds.fasta.gz
Maseb1_AssemblyScaffolds.fasta.gz
Melap1finSC_191_AssemblyScaffolds.fasta.gz
Melbi2_AssemblyScaffolds.fasta.gz
Melen1_AssemblyScaffolds.fasta.gz
Melli1_AssemblyScaffolds.fasta.gz
Melti1_AssemblyScaffolds.fasta.gz
Monha1_AssemblyScaffolds.fasta.gz
Morimp1_AssemblyScaffolds.fasta.gz
Mormul1_AssemblyScaffolds.fasta.gz
Mycosphaerella_graminicola.AssembledScaffolds.fasta.gz
Naifl1_AssemblyScaffolds.fasta.gz
Neosp1_AssemblyScaffolds.fasta.gz
Neurospora_tetrasperma_matA_v2.unmasked_nuclear_assembly.gz
Ophpi1_AssemblyScaffolds.fasta.gz
Orpsp1_1_AssemblyScaffolds.fasta.gz
Pacta1_2_AssemblyScaffolds.fasta.gz
Parbra1_AssemblyScaffolds.fasta.gz
Parsp1_AssemblyScaffolds.fasta.gz
Penra1_AssemblyScaffolds.fasta.gz
Penth1_AssemblyScaffolds.fasta.gz
Phaart1_AssemblyScaffolds.fasta.gz
Phcit1_AssemblyScaffolds.fasta.gz
Phcitr1_AssemblyScaffolds.fasta.gz
Phocon1_AssemblyScaffolds.fasta.gz
Phycit1_AssemblyScaffolds.fasta.gz
Phycomyces_blakesleeanus_v2_scaffolds.fasta.gz
Pilano1_AssemblyScaffolds.fasta.gz
Pilidi1_AssemblyScaffolds.fasta.gz
Pilidi1_MitoAssemblyScaffolds.fasta.gz
PirE2_1_AssemblyScaffolds.fasta.gz
Pirfi3_AssemblyScaffolds.fasta.gz
PleosPC9_1_Assembly_scaffolds.fasta.gz
Plesi1_AssemblyScaffolds.fasta.gz
Polci1_AssemblyScaffolds.fasta.gz
Pucgr2_AssemblyScaffolds.fasta.gz
Pucst_PST78_1_AssemblyScaffolds.fasta.gz
Rhich1_AssemblyScaffolds.fasta.gz
Rhihy1_AssemblyScaffolds.fasta.gz
Rhimi_ATCC11559_1_AssemblyScaffolds.fasta.gz
Rhior3_scaffolds.fasta.gz
Rhoba1_1_AssemblyScaffolds.fasta.gz
Rhobut1_1_AssemblyScaffolds.fasta.gz
Rhyru1_1_AssemblyScaffolds.fasta.gz
Ricfib1_AssemblyScaffolds.fasta.gz
Sacpr1_AssemblyScaffolds.fasta.gz
Sebve1_AssemblyScaffolds.fasta.gz
Settur1_AssemblyScaffolds.fasta.gz
Sodal1_AssemblyScaffolds.fasta.gz
Spifus1_AssemblyScaffolds.fasta.gz
Spopa1_AssemblyScaffolds.fasta.gz
Symat1_AssemblyScaffolds.fasta.gz
Synfus1_AssemblyScaffolds.fasta.gz
Terbo2_AssemblyScaffolds.fasta.gz
Thaele1_AssemblyScaffolds.fasta.gz
Theau1_AssemblyScaffolds.fasta.gz
ThoPMI491_1_AssemblyScaffolds.fasta.gz
Tilalb1_AssemblyScaffolds.fasta.gz
Torra1_AssemblyScaffolds.fasta.gz
Trapub1_AssemblyScaffolds.fasta.gz
Trepe1_AssemblyScaffolds.fasta.gz
Trias1_AssemblyScaffolds.fasta.gz
TrireRUTC30_1_AssemblyScaffolds.fasta.gz
Trisp1_AssemblyScaffolds.fasta.gz
Ustsp1_AssemblyScaffolds.fasta.gz
Volvo1_AssemblyScaffolds.fasta.gz
Xylhyp1_AssemblyScaffolds.fasta.gz
Zoprh1_AssemblyScaffolds.fasta.gz
Zycmex1_AssemblyScaffolds.fasta.gz
Zyghet1_AssemblyScaffolds.fasta.gz
```
Copy over files
``` 
cd /mnt/work/ref_genome/
mkdir nucmer_k31
cd nucmer_k31
while read inline
do
  cp ~/fungal_genomes/JGI_1KFGs/${inline} .
done < JGI_files_k31.txt
```

```
cd /mnt/work/ref_genome/nucmer_k31
touch other_file_names_k31.txt
while read inline
do
  grep -l "${inline}" ~/fungal_sigs/*sig
done < other_files_k31.txt
```
Output:
```
/home/ubuntu/fungal_sigs/GCA_001914275.1_FSWF8-bin-4_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCF_001642055.1_Altal1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001936605.2_ASM193660v2_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001676635.1_ASM167663v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001678115.1_ASM167811v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000410735.1_Opiceae_UAMHv01_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000785685.1_ASM78568v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000816005.1_KF-25_v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001276145.1_ASM127614v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001443065.1_ASM144306v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001661475.1_Canca1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001707075.1_ASM170707v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001442715.1_ASM144271v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000597685.1_Rir_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000697215.1_CunBer175-1.0_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001691765.1_ASM169176v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000166955.1_ASM16695v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000182075.1_M22_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000151525.2_P_triticina_1_1_V2_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001644785.1_ASM164478v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000222875.2_PerIpo_1.0_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001706785.1_PuccTri106_1.0_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000286035.1_ASM28603v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000190255.1_ScFostersB_v01_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_001179745.1_Cla_fum_l100.fa_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_000150085.1_ASM15008v1_genomic.fna.gz.sig
/home/ubuntu/fungal_sigs/GCA_900101485.1_version_1_genomic.fna.gz.sig
```
Edit to remove file path, and find file in fungal genome folder
```
GCA_001914275.1_FSWF8-bin-4_genomic.fna.gz
GCF_001642055.1_Altal1_genomic.fna.gz
GCA_001936605.2_ASM193660v2_genomic.fna.gz
GCA_001676635.1_ASM167663v1_genomic.fna.gz
GCA_001678115.1_ASM167811v1_genomic.fna.gz
GCA_000410735.1_Opiceae_UAMHv01_genomic.fna.gz
GCA_000785685.1_ASM78568v1_genomic.fna.gz
GCA_000816005.1_KF-25_v1_genomic.fna.gz
GCA_001276145.1_ASM127614v1_genomic.fna.gz
GCA_001443065.1_ASM144306v1_genomic.fna.gz
GCA_001661475.1_Canca1_genomic.fna.gz
GCA_001707075.1_ASM170707v1_genomic.fna.gz
GCA_001442715.1_ASM144271v1_genomic.fna.gz
GCA_000597685.1_Rir_genomic.fna.gz
GCA_000697215.1_CunBer175-1.0_genomic.fna.gz
GCA_001691765.1_ASM169176v1_genomic.fna.gz
GCA_000166955.1_ASM16695v1_genomic.fna.gz
GCA_000182075.1_M22_genomic.fna.gz
GCA_000151525.2_P_triticina_1_1_V2_genomic.fna.gz
GCA_001644785.1_ASM164478v1_genomic.fna.gz
GCA_000222875.2_PerIpo_1.0_genomic.fna.gz
GCA_001706785.1_PuccTri106_1.0_genomic.fna.gz
GCA_000286035.1_ASM28603v1_genomic.fna.gz
GCA_000190255.1_ScFostersB_v01_genomic.fna.gz
GCA_001179745.1_Cla_fum_l100.fa_genomic.fna.gz
GCA_000150085.1_ASM15008v1_genomic.fna.gz
GCA_900101485.1_version_1_genomic.fna.gz
```
Copy in files
```
touch other_file_paths_k31.txt
while read inline
do
  find ~/fungal_genomes/ -name $inline >> other_file_paths_k31.txt
done < other_file_names_k31.txt
  
while read inline
do
  cp ${inline} .
done < other_file_paths_k31.txt
```

Concatenate all files into one fasta file. Unzip files first to check if headers contain information about samples.
```
gunzip *gz 
```
.fna files are fine. JGI files need scaffold names to be replaced before concatenation. 
```
for infile in *fasta
do 
  find $infile -type f -exec sed -i "s/Superscaffold/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/scaffold/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/Supercontig/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/contig/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/LG/${infile}/g" {} \;
  find $infile -type f -exec sed -i "s/sc_/${infile}_/g" {} \;
  find $infile -type f -exec sed -i "s/c_/${infile}_/g" {} \;
  find $infile -type f -exec sed -i "s/d_/${infile}_/g" {} \;
done
```

Concatenate
```
cat *a > k31_concat.fa
```

Run nucmer!
```
~/mummer-4.0.0beta/nucmer -h
~/mummer-4.0.0beta/nucmer ../Oe6.scaffolds.fa k31_concat.fa -p k31_concat_vs_Oe6scaffolds

