Quantify read counts with Salmon

Make sure to activate `pondenv`, which was created during the eelpond protocol.
```
source ~/pondenv/bin/activate
```

```
cd /mnt/work/eelpond/
mkdir -p quant
cd quant
ln -s /mnt/work/eelpond/assembly/trinity_out_dir/Trinity.fasta .
```

Index transcriptome
```
salmon index --index olea-fruit --transcripts Trinity.fasta --type quasi
```

Link to qc'd reads
```
ln -s /mnt/work/single/quality/*qc.fq.gz .
ls
```  

Run Salmon for single end data
```
for file in SRR*
  do
    salmon quant -i olea-fruit -p 2 -l IU -r <(gunzip -c ${file}) -o ${file}.quant
done
```

Run Salmon for paired-end data
```
ln -s /mnt/work/paired/quality/ERR1346597_1.qc.fq.gz . 
ln -s /mnt/work/paired/quality/ERR1346597_2.qc.fq.gz .

  salmon quant -i olea-fruit -p 2 -l IU -1 <(gunzip -c ERR1346597_1.qc.fq.gz) -2 <(gunzip -c ERR1346597_2.qc.fq.gz) -o ERR1346597.qc.fq.gz.quant
```

Produce parsed count data
```
curl -L -O https://raw.githubusercontent.com/dib-lab/eel-pond/master/gather-counts.py
python ./gather-counts.py
```
