Run SSUsearch on trimmed, abundance filtered RNA-seq reads to detect ribosomal RNA for 16s (and ITS?) sequences.

Link in qc'd abundance trimmed reads (see preprocessing for trimmomatic sourmash folder for trim-low-abund for code on how this was done)
```
cd /mnt/work
mkdir SSUsearch
cd SSUsearch
ln -s /mnt/work/sourmash/trim-low-abund-v/*abundtrim .
```

Convert sequences to fasta format (format required for SSUsearch)
```
for infile in *abundtrim
do
  fastq_to_fasta -i $infile > ${infile}.fa
done
```

After SSUsearch finishes with the samples, gzip the .fa file that SSUsearch took as input to decrease the amount of space taken up on the device.
```
for infile in *fa
do
  gzip ${infile}
done
```


Remove `.fa` files when this is complete, as they take up too much space on the drive.
```
rm SRR1573503qc.fq.gz.abundtrim.fa.gz SRR1574328qc.fq.gz.abundtrim.fa.gz SRR1574719qc.fq.gz.abundtrim.fa.gz SRR1574772qc.fq.gz.abundtrim.fa.gz SRR4236909qc.fq.gz.abundtrim.fa.gz SRR4236910qc.fq.gz.abundtrim.fa.gz
```
